<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Exercises | Data Analysis &amp; Visualization using R (1)</title>
  <meta name="description" content="This gitbook accompanies the course Data Analysis &amp; Visualization using R 1 (DAVuR1) of the minor Bioinformatics for Life Science Students. It is organized by the Hanze University of Applied Science." />
  <meta name="generator" content="bookdown 0.11 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Exercises | Data Analysis &amp; Visualization using R (1)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This gitbook accompanies the course Data Analysis &amp; Visualization using R 1 (DAVuR1) of the minor Bioinformatics for Life Science Students. It is organized by the Hanze University of Applied Science." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Exercises | Data Analysis &amp; Visualization using R (1)" />
  
  <meta name="twitter:description" content="This gitbook accompanies the course Data Analysis &amp; Visualization using R 1 (DAVuR1) of the minor Bioinformatics for Life Science Students. It is organized by the Hanze University of Applied Science." />
  

<meta name="author" content="Michiel Noback" />


<meta name="date" content="2020-05-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="exploratory-data-analysis.html">
<link rel="next" href="exercise-solutions.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DAVuR eBook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Getting started</a></li>
<li class="chapter" data-level="2" data-path="toolbox.html"><a href="toolbox.html"><i class="fa fa-check"></i><b>2</b> The toolbox</a><ul>
<li class="chapter" data-level="2.1" data-path="toolbox.html"><a href="toolbox.html#embarking-on-data-science"><i class="fa fa-check"></i><b>2.1</b> Embarking on Data Science</a></li>
<li class="chapter" data-level="2.2" data-path="toolbox.html"><a href="toolbox.html#why-do-statistical-programming"><i class="fa fa-check"></i><b>2.2</b> Why do statistical programming?</a><ul>
<li class="chapter" data-level="" data-path="toolbox.html"><a href="toolbox.html#overview-of-the-toolbox"><i class="fa fa-check"></i>Overview of the toolbox</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="toolbox.html"><a href="toolbox.html#tool-1-the-r-programming-language"><i class="fa fa-check"></i><b>2.3</b> Tool 1: The R programming language</a></li>
<li class="chapter" data-level="2.4" data-path="toolbox.html"><a href="toolbox.html#tool-2-rstudio-as-development-environment"><i class="fa fa-check"></i><b>2.4</b> Tool 2: RStudio as development environment</a></li>
<li class="chapter" data-level="2.5" data-path="toolbox.html"><a href="toolbox.html#tool-3-r-markdown"><i class="fa fa-check"></i><b>2.5</b> Tool 3: R Markdown</a></li>
<li class="chapter" data-level="2.6" data-path="toolbox.html"><a href="toolbox.html#resources"><i class="fa fa-check"></i><b>2.6</b> Resources</a><ul>
<li class="chapter" data-level="" data-path="toolbox.html"><a href="toolbox.html#swirl-course"><i class="fa fa-check"></i>Swirl course</a></li>
<li class="chapter" data-level="" data-path="toolbox.html"><a href="toolbox.html#data-files"><i class="fa fa-check"></i>Data Files</a></li>
<li class="chapter" data-level="" data-path="toolbox.html"><a href="toolbox.html#web-resources-and-references"><i class="fa fa-check"></i>Web resources and references</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-r-coding.html"><a href="basic-r-coding.html"><i class="fa fa-check"></i><b>3</b> Basic R - coding</a><ul>
<li class="chapter" data-level="3.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#first-look-at-vectors-fuctions-and-variables"><i class="fa fa-check"></i><b>3.1</b> First look at vectors, fuctions and variables</a><ul>
<li class="chapter" data-level="3.1.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#doing-math-in-the-console"><i class="fa fa-check"></i><b>3.1.1</b> Doing Math in the console</a></li>
<li class="chapter" data-level="3.1.2" data-path="basic-r-coding.html"><a href="basic-r-coding.html#an-expression-dissected"><i class="fa fa-check"></i><b>3.1.2</b> An expression dissected</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="basic-r-coding.html"><a href="basic-r-coding.html#functions"><i class="fa fa-check"></i><b>3.2</b> Functions</a><ul>
<li class="chapter" data-level="3.2.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#getting-help-on-a-function"><i class="fa fa-check"></i><b>3.2.1</b> Getting help on a function</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="basic-r-coding.html"><a href="basic-r-coding.html#variables"><i class="fa fa-check"></i><b>3.3</b> Variables</a></li>
<li class="chapter" data-level="3.4" data-path="basic-r-coding.html"><a href="basic-r-coding.html#vectors"><i class="fa fa-check"></i><b>3.4</b> Vectors</a><ul>
<li class="chapter" data-level="3.4.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#r-is-completely-vector-based"><i class="fa fa-check"></i><b>3.4.1</b> R is completely vector-based</a></li>
<li class="chapter" data-level="3.4.2" data-path="basic-r-coding.html"><a href="basic-r-coding.html#five-datatype-that-live-in-vectors"><i class="fa fa-check"></i><b>3.4.2</b> Five datatype that live in vectors</a></li>
<li class="chapter" data-level="3.4.3" data-path="basic-r-coding.html"><a href="basic-r-coding.html#creating-vectors"><i class="fa fa-check"></i><b>3.4.3</b> Creating vectors</a></li>
<li class="chapter" data-level="3.4.4" data-path="basic-r-coding.html"><a href="basic-r-coding.html#vector-fiddling"><i class="fa fa-check"></i><b>3.4.4</b> Vector fiddling</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="basic-r-coding.html"><a href="basic-r-coding.html#other-operators"><i class="fa fa-check"></i><b>3.5</b> Other operators</a><ul>
<li class="chapter" data-level="3.5.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#logical-operators"><i class="fa fa-check"></i><b>3.5.1</b> Logical operators</a></li>
<li class="chapter" data-level="3.5.2" data-path="basic-r-coding.html"><a href="basic-r-coding.html#modulo"><i class="fa fa-check"></i><b>3.5.2</b> Modulo: <code>%%</code></a></li>
<li class="chapter" data-level="3.5.3" data-path="basic-r-coding.html"><a href="basic-r-coding.html#integer-division-and-rounding"><i class="fa fa-check"></i><b>3.5.3</b> Integer division <code>%/%</code> and rounding</a></li>
<li class="chapter" data-level="3.5.4" data-path="basic-r-coding.html"><a href="basic-r-coding.html#the-in-operator"><i class="fa fa-check"></i><b>3.5.4</b> The <code>%in%</code> operator</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="basic-r-coding.html"><a href="basic-r-coding.html#vector-creation-methods"><i class="fa fa-check"></i><b>3.6</b> Vector creation methods</a></li>
<li class="chapter" data-level="3.7" data-path="basic-r-coding.html"><a href="basic-r-coding.html#selecting-vector-elements"><i class="fa fa-check"></i><b>3.7</b> Selecting vector elements</a></li>
<li class="chapter" data-level="3.8" data-path="basic-r-coding.html"><a href="basic-r-coding.html#some-coding-style-rules-rules-for-writing-code"><i class="fa fa-check"></i><b>3.8</b> Some coding style rules rules for writing code</a></li>
<li class="chapter" data-level="3.9" data-path="basic-r-coding.html"><a href="basic-r-coding.html#the-best-keyboard-shortcuts-for-rstudio"><i class="fa fa-check"></i><b>3.9</b> The best keyboard shortcuts for RStudio</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html"><i class="fa fa-check"></i><b>4</b> Basic R - plotting</a><ul>
<li class="chapter" data-level="4.1" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#basic-embedded-plot-types"><i class="fa fa-check"></i><b>4.1</b> Basic embedded plot types</a><ul>
<li class="chapter" data-level="4.1.1" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#scatter-and-line-plots"><i class="fa fa-check"></i><b>4.1.1</b> Scatter and line plots</a></li>
<li class="chapter" data-level="4.1.2" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#barplots"><i class="fa fa-check"></i><b>4.1.2</b> Barplots</a></li>
<li class="chapter" data-level="4.1.3" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#histograms"><i class="fa fa-check"></i><b>4.1.3</b> Histograms</a></li>
<li class="chapter" data-level="4.1.4" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#density-plot-as-alternative-to-hist"><i class="fa fa-check"></i><b>4.1.4</b> Density plot as alternative to <code>hist()</code></a></li>
<li class="chapter" data-level="4.1.5" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#boxplots"><i class="fa fa-check"></i><b>4.1.5</b> Boxplots</a></li>
<li class="chapter" data-level="4.1.6" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#adding-more-data-and-a-legend"><i class="fa fa-check"></i><b>4.1.6</b> Adding more data and a legend</a></li>
<li class="chapter" data-level="4.1.7" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#helper-lines-and-lm"><i class="fa fa-check"></i><b>4.1.7</b> Helper lines and <code>lm()</code></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#graphical-parameters-to-plot"><i class="fa fa-check"></i><b>4.2</b> Graphical parameters to <code>plot()</code></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html"><i class="fa fa-check"></i><b>5</b> Complex Datatypes and File Reading</a><ul>
<li class="chapter" data-level="5.1" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#matrices-are-vectors-with-dimensions"><i class="fa fa-check"></i><b>5.1</b> Matrices are vectors with dimensions</a></li>
<li class="chapter" data-level="5.2" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#factors-nominal-ordinal-scales"><i class="fa fa-check"></i><b>5.2</b> Factors: Nominal &amp; Ordinal scales</a></li>
<li class="chapter" data-level="5.3" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#lists"><i class="fa fa-check"></i><b>5.3</b> Lists</a></li>
<li class="chapter" data-level="5.4" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#dataframes"><i class="fa fa-check"></i><b>5.4</b> Dataframes</a><ul>
<li class="chapter" data-level="5.4.1" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#selections-on-dataframes"><i class="fa fa-check"></i><b>5.4.1</b> Selections on dataframes</a></li>
<li class="chapter" data-level="5.4.2" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#read-from-file-using-read.table"><i class="fa fa-check"></i><b>5.4.2</b> Read from file using <code>read.table()</code></a></li>
<li class="chapter" data-level="5.4.3" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#dataframe-manipulations"><i class="fa fa-check"></i><b>5.4.3</b> Dataframe manipulations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="functions-1.html"><a href="functions-1.html"><i class="fa fa-check"></i><b>6</b> Functions</a><ul>
<li class="chapter" data-level="6.1" data-path="functions-1.html"><a href="functions-1.html#dealing-with-nas"><i class="fa fa-check"></i><b>6.1</b> Dealing with NAs</a></li>
<li class="chapter" data-level="6.2" data-path="functions-1.html"><a href="functions-1.html#descriptive-statistics"><i class="fa fa-check"></i><b>6.2</b> Descriptive statistics</a></li>
<li class="chapter" data-level="6.3" data-path="functions-1.html"><a href="functions-1.html#general-purpose-functions"><i class="fa fa-check"></i><b>6.3</b> General purpose functions</a></li>
<li class="chapter" data-level="6.4" data-path="functions-1.html"><a href="functions-1.html#convert-numeric-vector-to-factor-cut"><i class="fa fa-check"></i><b>6.4</b> Convert numeric vector to factor: <code>cut()</code></a></li>
<li class="chapter" data-level="6.5" data-path="functions-1.html"><a href="functions-1.html#file-io-revisited"><i class="fa fa-check"></i><b>6.5</b> File I/O revisited</a></li>
<li class="chapter" data-level="6.6" data-path="functions-1.html"><a href="functions-1.html#pattern-matching"><i class="fa fa-check"></i><b>6.6</b> Pattern matching</a><ul>
<li class="chapter" data-level="6.6.1" data-path="functions-1.html"><a href="functions-1.html#regex-syntax"><i class="fa fa-check"></i><b>6.6.1</b> Regex syntax</a></li>
<li class="chapter" data-level="6.6.2" data-path="functions-1.html"><a href="functions-1.html#quantifiers"><i class="fa fa-check"></i><b>6.6.2</b> Quantifiers</a></li>
<li class="chapter" data-level="6.6.3" data-path="functions-1.html"><a href="functions-1.html#some-examples"><i class="fa fa-check"></i><b>6.6.3</b> Some examples</a></li>
<li class="chapter" data-level="6.6.4" data-path="functions-1.html"><a href="functions-1.html#anchoring"><i class="fa fa-check"></i><b>6.6.4</b> Anchoring</a></li>
<li class="chapter" data-level="6.6.5" data-path="functions-1.html"><a href="functions-1.html#metacharacters-special-character-classes"><i class="fa fa-check"></i><b>6.6.5</b> Metacharacters: Special character classes</a></li>
<li class="chapter" data-level="6.6.6" data-path="functions-1.html"><a href="functions-1.html#locating-and-extracting"><i class="fa fa-check"></i><b>6.6.6</b> Locating and Extracting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="scripting.html"><a href="scripting.html"><i class="fa fa-check"></i><b>7</b> Scripting</a><ul>
<li class="chapter" data-level="7.1" data-path="scripting.html"><a href="scripting.html#introduction"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="scripting.html"><a href="scripting.html#flow-control"><i class="fa fa-check"></i><b>7.2</b> Flow control</a><ul>
<li class="chapter" data-level="7.2.1" data-path="scripting.html"><a href="scripting.html#conditional-execution-with-ifelse"><i class="fa fa-check"></i><b>7.2.1</b> Conditional execution with <code>if/else</code></a></li>
<li class="chapter" data-level="7.2.2" data-path="scripting.html"><a href="scripting.html#iteration-with-for"><i class="fa fa-check"></i><b>7.2.2</b> Iteration with <code>for(){}</code></a></li>
<li class="chapter" data-level="7.2.3" data-path="scripting.html"><a href="scripting.html#conditional-iteration-with-while"><i class="fa fa-check"></i><b>7.2.3</b> Conditional iteration with <code>while(){}</code></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="scripting.html"><a href="scripting.html#creating-functions"><i class="fa fa-check"></i><b>7.3</b> Creating functions</a><ul>
<li class="chapter" data-level="7.3.1" data-path="scripting.html"><a href="scripting.html#errors-and-warnings"><i class="fa fa-check"></i><b>7.3.1</b> Errors and warnings</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="scripting.html"><a href="scripting.html#scripting-1"><i class="fa fa-check"></i><b>7.4</b> Scripting</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html"><i class="fa fa-check"></i><b>8</b> Dataframe manipulations</a><ul>
<li class="chapter" data-level="8.1" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#the-apply-family-of-functions"><i class="fa fa-check"></i><b>8.1</b> The <code>apply()</code> family of functions</a><ul>
<li class="chapter" data-level="8.1.1" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#apply-apply-functions-over-array-margins"><i class="fa fa-check"></i><b>8.1.1</b> <code>apply()</code>: Apply Functions Over Array Margins</a></li>
<li class="chapter" data-level="8.1.2" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#lapply-apply-a-function-over-a-list-or-vector"><i class="fa fa-check"></i><b>8.1.2</b> <code>lapply()</code>: Apply a Function over a List or Vector</a></li>
<li class="chapter" data-level="8.1.3" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#sapply-apply-a-function-over-a-list-or-vector-and-simplify"><i class="fa fa-check"></i><b>8.1.3</b> <code>sapply()</code>: Apply a Function over a List or Vector and Simplify</a></li>
<li class="chapter" data-level="8.1.4" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#tapply-apply-a-function-over-a-ragged-array"><i class="fa fa-check"></i><b>8.1.4</b> <code>tapply()</code>: Apply a Function Over a Ragged Array</a></li>
<li class="chapter" data-level="8.1.5" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#split-divide-into-groups-and-reassemble"><i class="fa fa-check"></i><b>8.1.5</b> <code>split()</code>: Divide into Groups and Reassemble</a></li>
<li class="chapter" data-level="8.1.6" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#aggregate-compute-summary-statistics-of-data-subsets"><i class="fa fa-check"></i><b>8.1.6</b> <code>aggregate()</code>: Compute Summary Statistics of Data Subsets</a></li>
<li class="chapter" data-level="8.1.7" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#many-roads-lead-to-rome"><i class="fa fa-check"></i><b>8.1.7</b> Many roads lead to Rome</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#usecases"><i class="fa fa-check"></i><b>8.2</b> Example Use Cases</a><ul>
<li class="chapter" data-level="8.2.1" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#dfselection"><i class="fa fa-check"></i><b>8.2.1</b> Dataframe Selections</a></li>
<li class="chapter" data-level="8.2.2" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#apply"><i class="fa fa-check"></i><b>8.2.2</b> Apply</a></li>
<li class="chapter" data-level="8.2.3" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#embeddeddf"><i class="fa fa-check"></i><b>8.2.3</b> Processing Embedded Dataframes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>9</b> Exploratory Data Analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#introduction-1"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#eda-of-the-yeast-dataset"><i class="fa fa-check"></i><b>9.2</b> EDA of the Yeast dataset</a><ul>
<li class="chapter" data-level="9.2.1" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#introduction-2"><i class="fa fa-check"></i><b>9.2.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2.2" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#data-loading-and-prepping"><i class="fa fa-check"></i><b>9.2.2</b> Data loading and prepping</a></li>
<li class="chapter" data-level="9.2.3" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#data-verification"><i class="fa fa-check"></i><b>9.2.3</b> Data verification</a></li>
<li class="chapter" data-level="9.2.4" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#attribute-summaries"><i class="fa fa-check"></i><b>9.2.4</b> Attribute summaries</a></li>
<li class="chapter" data-level="9.2.5" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#attribute-relationships"><i class="fa fa-check"></i><b>9.2.5</b> Attribute relationships</a></li>
<li class="chapter" data-level="9.2.6" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#correlations-to-the-dependent-variable"><i class="fa fa-check"></i><b>9.2.6</b> Correlations to the dependent variable</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>10</b> Exercises</a><ul>
<li class="chapter" data-level="10.1" data-path="exercises.html"><a href="exercises.html#toolbox-1"><i class="fa fa-check"></i><b>10.1</b> Toolbox</a><ul>
<li class="chapter" data-level="10.1.1" data-path="exercises.html"><a href="exercises.html#get-set-up"><i class="fa fa-check"></i><b>10.1.1</b> Get set up</a></li>
<li class="chapter" data-level="10.1.2" data-path="exercises.html"><a href="exercises.html#resume"><i class="fa fa-check"></i><b>10.1.2</b> Résumé</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="exercises.html"><a href="exercises.html#basic-r"><i class="fa fa-check"></i><b>10.2</b> Basic R</a><ul>
<li class="chapter" data-level="10.2.1" data-path="exercises.html"><a href="exercises.html#math-in-the-console"><i class="fa fa-check"></i><b>10.2.1</b> Math in the console</a></li>
<li class="chapter" data-level="10.2.2" data-path="exercises.html"><a href="exercises.html#first-look-at-functions"><i class="fa fa-check"></i><b>10.2.2</b> First look at functions</a></li>
<li class="chapter" data-level="10.2.3" data-path="exercises.html"><a href="exercises.html#variables-1"><i class="fa fa-check"></i><b>10.2.3</b> Variables</a></li>
<li class="chapter" data-level="10.2.4" data-path="exercises.html"><a href="exercises.html#vectors-1"><i class="fa fa-check"></i><b>10.2.4</b> Vectors</a></li>
<li class="chapter" data-level="10.2.5" data-path="exercises.html"><a href="exercises.html#stair-walking-and-heart-rate"><i class="fa fa-check"></i><b>10.2.5</b> Stair walking and heart rate</a></li>
<li class="chapter" data-level="10.2.6" data-path="exercises.html"><a href="exercises.html#more-subjects"><i class="fa fa-check"></i><b>10.2.6</b> More subjects</a></li>
<li class="chapter" data-level="10.2.7" data-path="exercises.html"><a href="exercises.html#chickens-on-a-diet"><i class="fa fa-check"></i><b>10.2.7</b> Chickens on a diet</a></li>
<li class="chapter" data-level="10.2.8" data-path="exercises.html"><a href="exercises.html#chicken-bar-plot"><i class="fa fa-check"></i><b>10.2.8</b> Chicken bar plot</a></li>
<li class="chapter" data-level="10.2.9" data-path="exercises.html"><a href="exercises.html#discoveries"><i class="fa fa-check"></i><b>10.2.9</b> Discoveries</a></li>
<li class="chapter" data-level="10.2.10" data-path="exercises.html"><a href="exercises.html#lung-cancer"><i class="fa fa-check"></i><b>10.2.10</b> Lung cancer</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="exercises.html"><a href="exercises.html#complex-datatypes"><i class="fa fa-check"></i><b>10.3</b> Complex datatypes</a><ul>
<li class="chapter" data-level="10.3.1" data-path="exercises.html"><a href="exercises.html#creating-factors"><i class="fa fa-check"></i><b>10.3.1</b> Creating factors</a></li>
<li class="chapter" data-level="10.3.2" data-path="exercises.html"><a href="exercises.html#a-dictionary-with-a-named-vector"><i class="fa fa-check"></i><b>10.3.2</b> A dictionary with a named vector</a></li>
<li class="chapter" data-level="10.3.3" data-path="exercises.html"><a href="exercises.html#protein-concentrations-with-lowry"><i class="fa fa-check"></i><b>10.3.3</b> Protein concentrations with Lowry</a></li>
<li class="chapter" data-level="10.3.4" data-path="exercises.html"><a href="exercises.html#hplc-data"><i class="fa fa-check"></i><b>10.3.4</b> HPLC data</a></li>
<li class="chapter" data-level="10.3.5" data-path="exercises.html"><a href="exercises.html#airquality"><i class="fa fa-check"></i><b>10.3.5</b> Airquality</a></li>
<li class="chapter" data-level="10.3.6" data-path="exercises.html"><a href="exercises.html#file-reading-practice"><i class="fa fa-check"></i><b>10.3.6</b> File reading practice</a></li>
<li class="chapter" data-level="10.3.7" data-path="exercises.html"><a href="exercises.html#bird-observations"><i class="fa fa-check"></i><b>10.3.7</b> Bird observations</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="exercises.html"><a href="exercises.html#regular-expressions"><i class="fa fa-check"></i><b>10.4</b> Regular Expressions</a><ul>
<li class="chapter" data-level="10.4.1" data-path="exercises.html"><a href="exercises.html#restriction-enzymes-1"><i class="fa fa-check"></i><b>10.4.1</b> Restriction enzymes</a></li>
<li class="chapter" data-level="10.4.2" data-path="exercises.html"><a href="exercises.html#prosite-patterns-1"><i class="fa fa-check"></i><b>10.4.2</b> Prosite Patterns</a></li>
<li class="chapter" data-level="10.4.3" data-path="exercises.html"><a href="exercises.html#fasta-headers"><i class="fa fa-check"></i><b>10.4.3</b> Fasta Headers</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="exercises.html"><a href="exercises.html#scripting-2"><i class="fa fa-check"></i><b>10.5</b> Scripting</a><ul>
<li class="chapter" data-level="10.5.1" data-path="exercises.html"><a href="exercises.html#illegal-reproductions"><i class="fa fa-check"></i><b>10.5.1</b> Illegal reproductions</a></li>
<li class="chapter" data-level="10.5.2" data-path="exercises.html"><a href="exercises.html#interquantile-ranges"><i class="fa fa-check"></i><b>10.5.2</b> Interquantile ranges</a></li>
<li class="chapter" data-level="10.5.3" data-path="exercises.html"><a href="exercises.html#vector-distance"><i class="fa fa-check"></i><b>10.5.3</b> Vector distance</a></li>
<li class="chapter" data-level="10.5.4" data-path="exercises.html"><a href="exercises.html#gc-percentage-of-dna"><i class="fa fa-check"></i><b>10.5.4</b> G/C percentage of DNA</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="exercises.html"><a href="exercises.html#function-apply-and-its-relatives"><i class="fa fa-check"></i><b>10.6</b> Function <code>apply</code> and its relatives</a><ul>
<li class="chapter" data-level="10.6.1" data-path="exercises.html"><a href="exercises.html#whale-selenium"><i class="fa fa-check"></i><b>10.6.1</b> Whale selenium</a></li>
<li class="chapter" data-level="10.6.2" data-path="exercises.html"><a href="exercises.html#chickweight"><i class="fa fa-check"></i><b>10.6.2</b> ChickWeight</a></li>
<li class="chapter" data-level="10.6.3" data-path="exercises.html"><a href="exercises.html#food-constituents"><i class="fa fa-check"></i><b>10.6.3</b> Food constituents</a></li>
<li class="chapter" data-level="10.6.4" data-path="exercises.html"><a href="exercises.html#urine-properties"><i class="fa fa-check"></i><b>10.6.4</b> Urine properties</a></li>
<li class="chapter" data-level="10.6.5" data-path="exercises.html"><a href="exercises.html#bird-observations-revisited"><i class="fa fa-check"></i><b>10.6.5</b> Bird observations revisited</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="exercise-solutions.html"><a href="exercise-solutions.html"><i class="fa fa-check"></i><b>11</b> Exercise solutions</a><ul>
<li class="chapter" data-level="11.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#toolbox-2"><i class="fa fa-check"></i><b>11.1</b> Toolbox</a><ul>
<li class="chapter" data-level="11.1.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#get-set-up-1"><i class="fa fa-check"></i><b>11.1.1</b> Get set up</a></li>
<li class="chapter" data-level="11.1.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#resume-1"><i class="fa fa-check"></i><b>11.1.2</b> Résumé</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#basic-r-1"><i class="fa fa-check"></i><b>11.2</b> Basic R</a><ul>
<li class="chapter" data-level="11.2.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#math-in-the-console-1"><i class="fa fa-check"></i><b>11.2.1</b> Math in the console</a></li>
<li class="chapter" data-level="11.2.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#first-look-at-functions-1"><i class="fa fa-check"></i><b>11.2.2</b> First look at functions</a></li>
<li class="chapter" data-level="11.2.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#variables-2"><i class="fa fa-check"></i><b>11.2.3</b> Variables</a></li>
<li class="chapter" data-level="11.2.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#vectors-2"><i class="fa fa-check"></i><b>11.2.4</b> Vectors</a></li>
<li class="chapter" data-level="11.2.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#stair-walking-and-heart-rate-1"><i class="fa fa-check"></i><b>11.2.5</b> Stair walking and heart rate</a></li>
<li class="chapter" data-level="11.2.6" data-path="exercise-solutions.html"><a href="exercise-solutions.html#more-subjects-1"><i class="fa fa-check"></i><b>11.2.6</b> More subjects</a></li>
<li class="chapter" data-level="11.2.7" data-path="exercise-solutions.html"><a href="exercise-solutions.html#chickens-on-a-diet-1"><i class="fa fa-check"></i><b>11.2.7</b> Chickens on a diet</a></li>
<li class="chapter" data-level="11.2.8" data-path="exercise-solutions.html"><a href="exercise-solutions.html#chicken-bar-plot-1"><i class="fa fa-check"></i><b>11.2.8</b> Chicken bar plot</a></li>
<li class="chapter" data-level="11.2.9" data-path="exercise-solutions.html"><a href="exercise-solutions.html#discoveries-1"><i class="fa fa-check"></i><b>11.2.9</b> Discoveries</a></li>
<li class="chapter" data-level="11.2.10" data-path="exercise-solutions.html"><a href="exercise-solutions.html#lung-cancer-1"><i class="fa fa-check"></i><b>11.2.10</b> Lung cancer</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#complex-datatypes-1"><i class="fa fa-check"></i><b>11.3</b> Complex datatypes</a><ul>
<li class="chapter" data-level="11.3.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#creating-factors-1"><i class="fa fa-check"></i><b>11.3.1</b> Creating factors</a></li>
<li class="chapter" data-level="11.3.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#a-dictionary-with-a-named-vector-1"><i class="fa fa-check"></i><b>11.3.2</b> A dictionary with a named vector</a></li>
<li class="chapter" data-level="11.3.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#protein-concentrations-with-lowry-1"><i class="fa fa-check"></i><b>11.3.3</b> Protein concentrations with Lowry</a></li>
<li class="chapter" data-level="11.3.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#hplc-data-1"><i class="fa fa-check"></i><b>11.3.4</b> HPLC data</a></li>
<li class="chapter" data-level="11.3.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#airquality-1"><i class="fa fa-check"></i><b>11.3.5</b> Airquality</a></li>
<li class="chapter" data-level="11.3.6" data-path="exercise-solutions.html"><a href="exercise-solutions.html#file-reading-practice-1"><i class="fa fa-check"></i><b>11.3.6</b> File reading practice</a></li>
<li class="chapter" data-level="11.3.7" data-path="exercise-solutions.html"><a href="exercise-solutions.html#bird-observations-1"><i class="fa fa-check"></i><b>11.3.7</b> Bird observations</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#regular-expressions-1"><i class="fa fa-check"></i><b>11.4</b> Regular Expressions</a><ul>
<li class="chapter" data-level="11.4.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#restriction-enzymes-2"><i class="fa fa-check"></i><b>11.4.1</b> Restriction enzymes</a></li>
<li class="chapter" data-level="11.4.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#prosite-patterns-2"><i class="fa fa-check"></i><b>11.4.2</b> Prosite Patterns</a></li>
<li class="chapter" data-level="11.4.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#fasta-headers-1"><i class="fa fa-check"></i><b>11.4.3</b> Fasta Headers</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#scripting-3"><i class="fa fa-check"></i><b>11.5</b> Scripting</a><ul>
<li class="chapter" data-level="11.5.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#illegal-reproductions-1"><i class="fa fa-check"></i><b>11.5.1</b> Illegal reproductions</a></li>
<li class="chapter" data-level="11.5.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#interquantile-ranges-1"><i class="fa fa-check"></i><b>11.5.2</b> Interquantile ranges</a></li>
<li class="chapter" data-level="11.5.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#vector-distance-1"><i class="fa fa-check"></i><b>11.5.3</b> Vector distance</a></li>
<li class="chapter" data-level="11.5.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#gc-percentage-of-dna-1"><i class="fa fa-check"></i><b>11.5.4</b> G/C percentage of DNA</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="exercise-solutions.html"><a href="exercise-solutions.html#function-apply-and-its-relatives-1"><i class="fa fa-check"></i><b>11.6</b> Function <code>apply</code> and its relatives</a><ul>
<li class="chapter" data-level="11.6.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#whale-selenium-1"><i class="fa fa-check"></i><b>11.6.1</b> Whale selenium</a></li>
<li class="chapter" data-level="11.6.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#chickweight-1"><i class="fa fa-check"></i><b>11.6.2</b> ChickWeight</a></li>
<li class="chapter" data-level="11.6.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#food-constituents-1"><i class="fa fa-check"></i><b>11.6.3</b> Food constituents</a></li>
<li class="chapter" data-level="11.6.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#urine-properties-1"><i class="fa fa-check"></i><b>11.6.4</b> Urine properties</a></li>
<li class="chapter" data-level="11.6.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#bird-observations-revisited-1"><i class="fa fa-check"></i><b>11.6.5</b> Bird observations revisited</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis &amp; Visualization using R (1)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exercises" class="section level1">
<h1><span class="header-section-number">Chapter 10</span> Exercises</h1>
<p>This chapter only contains exercises. The solutions are in the next chapter which has a numbering parallel to this one. To work on the exercises it is probably best to create either an RMarkdown Notebook (File → New File → R Notebook) or an RMarkdown document (File → New File → R Markdown… → choose Document as type and give it a title). A Notebook has the advantage that you can toggle the visibility of the code, but it has fewer output options (no pdf or Word). I strongly suggest you play around with both.</p>
<p>This is a link to an <a href="figures/R_cheatsheet.pdf">R cheat sheet</a> that you could use for selecting the right function for a task.</p>
<p>If you want you can download this chapter as separate file and work in that. You can download it <a href="https://raw.githubusercontent.com/MichielNoback/davur1_gitbook/master/07-exercises.Rmd">here</a>. Before commencing you should put in a new Markdown header at the top of the file, with at least this contents:</p>
<pre><code>---
title: &quot;Exercise solutions DAVuR1&quot;
author: &quot;&lt;YOUR NAME&gt;&quot;
---</code></pre>
<div id="plotting-rules" class="section level4 unnumbered">
<h4>Plotting rules</h4>
<p>With all plots, take care to adhere to the rules regarding titles and other decorations. Tip: the site <a href="http://www.statmethods.net/">Quick-R</a> has nice detailed information with examples on the different plot types and their configuration. Especially the section <a href="http://www.statmethods.net/graphs/index.html">on plotting</a> is helpful for these assignments.</p>
</div>
<div id="toolbox-1" class="section level2">
<h2><span class="header-section-number">10.1</span> Toolbox</h2>
<div id="get-set-up" class="section level3">
<h3><span class="header-section-number">10.1.1</span> Get set up</h3>
<p>Install these tools on your own PC or laptop, in this order:</p>
<ol style="list-style-type: decimal">
<li>R itself <a href="https://cran.r-project.org/bin/windows/base/">https://cran.r-project.org/bin/windows/base/</a><br />
</li>
<li>RStudio <a href="https://rstudio.com/products/rstudio/download/">https://rstudio.com/products/rstudio/download/</a> (choose the free version of course)</li>
<li>[Optionally] If you want to generate pdf documents you should also install a Latex version: MikTeX or TinyTex for Windows or MacTeX on MacOS. If you want to keep things simple I suggest you stick to HTML and MS Word output (Word can also export to PDF).</li>
</ol>
</div>
<div id="resume" class="section level3">
<h3><span class="header-section-number">10.1.2</span> Résumé</h3>
<p>Create an R Markdown document called Résumé.Rmd and create your Curriculum Vitae using Markdown syntax. Use the R Markdown Reference Guide or Cheat Sheet or this <a href="https://guides.github.com/pdfs/markdown-cheatsheet-online.pdf">Markdown Cheatsheet</a> (you don’t need R for your résumé so markdown alone is enough).</p>
</div>
</div>
<div id="basic-r" class="section level2">
<h2><span class="header-section-number">10.2</span> Basic R</h2>
<div id="math-in-the-console" class="section level3">
<h3><span class="header-section-number">10.2.1</span> Math in the console</h3>
<p>In the console, calculate the following:</p>
<p><span class="math inline">\(31 + 11\)</span></p>
<p><span class="math inline">\(66 - 24\)</span></p>
<p><span class="math inline">\(\frac{126}{3}\)</span></p>
<p><span class="math inline">\(12^2\)</span></p>
<p><span class="math inline">\(\sqrt{256}\)</span></p>
<p><span class="math inline">\(\frac{3*(4+\sqrt{8})}{5^3}\)</span></p>
</div>
<div id="first-look-at-functions" class="section level3">
<h3><span class="header-section-number">10.2.2</span> First look at functions</h3>
<p><strong>A</strong><br />
View the help page for <code>paste()</code>. There are two variants of this function. Which? And what is the difference between them? Use both variants to generate <em>exactly</em> this message <code>&quot;welcome to R&quot;</code> from these arguments: <code>&quot;welcome &quot;, &quot;to &quot;, &quot;R&quot;</code></p>
<p><strong>B</strong><br />
What does the <code>abs</code> function do? What is returned by <code>abs(-20)</code> and what is <code>abs(20)</code>?</p>
<p><strong>C</strong><br />
What does the <code>c</code> function do? What is the difference in returned value (result) of <code>c()</code> when you combine either <code>1</code>, <code>3</code> and <code>&quot;a&quot;</code> as arguments, or <code>1</code>, <code>2</code> and <code>3</code>? Use the function <code>class()</code> to answer this.</p>
<p><strong>D</strong><br />
What does the function <code>install.packages()</code> do? Use it to install the package <code>RColorBrewer</code>. We’ll have a look at this package later on.</p>
</div>
<div id="variables-1" class="section level3">
<h3><span class="header-section-number">10.2.3</span> Variables</h3>
<p>Create three variables with the given values - x=20, y=10 and z=3. Next, calculate the following with these variables:</p>
<ol style="list-style-type: decimal">
<li><span class="math inline">\(x+y\)</span></li>
<li><span class="math inline">\(x^z\)</span></li>
<li><span class="math inline">\(q = x \times y \times z\)</span></li>
<li><span class="math inline">\(\sqrt{q}\)</span></li>
<li><span class="math inline">\(\frac{q}{\pi}\)</span> (pi is simply pi in R)</li>
<li><span class="math inline">\(\log_{10}{(x \times y)}\)</span></li>
</ol>
</div>
<div id="vectors-1" class="section level3">
<h3><span class="header-section-number">10.2.4</span> Vectors</h3>
<div id="circles" class="section level4 unnumbered">
<h4>Circles</h4>
<p>The circumference of a circle is <span class="math inline">\(2\pi\cdot r\)</span>, its surface <span class="math inline">\(4\pi \cdot r^2\)</span> and its volume <span class="math inline">\(4/3 \pi\cdot r^3\)</span>.
Given this vector of circle radiuses,</p>
<pre class="sourceCode r"><code class="sourceCode r">radiuses &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, pi, <span class="dv">4</span>)</code></pre>
<p><strong>A</strong><br />
Calculate their cirumference.</p>
<p><strong>B</strong><br />
Calculate their surface.</p>
<p><strong>C</strong><br />
Calculate their volume.</p>
</div>
<div id="creating-vectors-1" class="section level4 unnumbered">
<h4>Creating vectors</h4>
<p>Create the following vectors, <em>as efficiently as possible</em>. The functions <code>rep()</code>, <code>seq()</code>, <code>paste0()</code> and the colon operator <code>:</code> can be used, in any combination.</p>
<p><strong>A</strong><br />
<code>[1] 1 2 5 1 2 5</code></p>
<p><strong>B</strong><br />
<code>[1] 9 9 9 8 8 8 7 7 7 6 6 6 5 5 5</code></p>
<p><strong>C</strong><br />
<code>[1] 1 1 1 4 4 4 9 9 9 1 1 1 4 4 4 9 9 9</code></p>
<p><strong>D</strong><br />
<code>[1] &quot;1a&quot; &quot;2b&quot; &quot;3c&quot; &quot;4d&quot; &quot;5e&quot; &quot;1a&quot; &quot;2b&quot; &quot;3c&quot; &quot;4d&quot; &quot;5e&quot;</code></p>
<p><strong>E</strong><br />
<code>[1] &quot;0z&quot;   &quot;0.2y&quot; &quot;0.4x&quot; &quot;0.6w&quot; &quot;0.8v&quot; &quot;1u&quot;</code></p>
<p><strong>F</strong><br />
<code>[1] &quot;505&quot; &quot;404&quot; &quot;303&quot; &quot;202&quot; &quot;101&quot; &quot;000&quot;</code></p>
<p><strong>G [Challenge]</strong><br />
<code>[1] &quot;0.5A5.0&quot; &quot;0.4B4.0&quot; &quot;0.3C3.0&quot; &quot;0.2D2.0&quot; &quot;0.1E1.0&quot;</code></p>
</div>
</div>
<div id="stair-walking-and-heart-rate" class="section level3">
<h3><span class="header-section-number">10.2.5</span> Stair walking and heart rate</h3>
<p>The vectors below hold data for a staircase walking experiment. A subject of normal weight and height was asked to ascend a (long) stairs wearing a heart-rate monitor. The subjects’ heart was registered for different step heights. Create a <strong>line plot</strong> showing the dependence of heart rate (y axis) on stair height (x axis).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">#number of steps on the stairs</span>
stair_height &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">15</span>, <span class="dv">20</span>, <span class="dv">25</span>, <span class="dv">30</span>, <span class="dv">35</span>)
<span class="co">#heart rate after ascending the stairs</span>
heart_rate &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">66</span>, <span class="dv">65</span>, <span class="dv">67</span>, <span class="dv">69</span>, <span class="dv">73</span>, <span class="dv">79</span>, <span class="dv">86</span>, <span class="dv">97</span>)</code></pre>
</div>
<div id="more-subjects" class="section level3">
<h3><span class="header-section-number">10.2.6</span> More subjects</h3>
<p>The experiment from the previous question was extended with three more subjects. One of these subjects was like the first of normal weight, whereas the two others were obese. The data are given below. Create a single <strong>scatter plot</strong> with connector lines between the points showing the data for all four subjects. Give the normal-weighted subjects a green line and symbol and the obese subjects a red line and symbol.<br />
You can add new data series to a plot by using the <code>points(x, y)</code> function. Use the <code>ylim()</code> function to adjust the Y-axis range.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">#number of steps on the stairs</span>
stair_height &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">15</span>, <span class="dv">20</span>, <span class="dv">25</span>, <span class="dv">30</span>, <span class="dv">35</span>)
<span class="co">#heart rates for subjects with normal weight</span>
heart_rate_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">66</span>, <span class="dv">65</span>, <span class="dv">67</span>, <span class="dv">69</span>, <span class="dv">73</span>, <span class="dv">79</span>, <span class="dv">86</span>, <span class="dv">97</span>)
heart_rate_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">61</span>, <span class="dv">61</span>, <span class="dv">63</span>, <span class="dv">68</span>, <span class="dv">74</span>, <span class="dv">81</span>, <span class="dv">89</span>, <span class="dv">104</span>)
<span class="co">#heart rates for obese subjects</span>
heart_rate_<span class="dv">3</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">58</span>, <span class="dv">60</span>, <span class="dv">67</span>, <span class="dv">71</span>, <span class="dv">78</span>, <span class="dv">89</span>, <span class="dv">104</span>, <span class="dv">121</span>)
heart_rate_<span class="dv">4</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">69</span>, <span class="dv">73</span>, <span class="dv">77</span>, <span class="dv">83</span>, <span class="dv">88</span>, <span class="dv">96</span>, <span class="dv">102</span>, <span class="dv">127</span>)</code></pre>
</div>
<div id="chickens-on-a-diet" class="section level3">
<h3><span class="header-section-number">10.2.7</span> Chickens on a diet</h3>
<p>The body weights of chicks were measured at birth and every second day thereafter until day 20. They were also measured on day 21. In the experiment there were four groups of chicks on different protein diets. Here are the data for the first four chicks. Chick one and two were on diet 1 and chick three and four were on diet 2. Create a single line plot showing the data for all four chicks. Give each chick its own color.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co"># chick weight data</span>
time &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">8</span>, <span class="dv">10</span>, <span class="dv">12</span>, <span class="dv">14</span>, <span class="dv">16</span>, <span class="dv">18</span>, <span class="dv">20</span>, <span class="dv">21</span>)
chick_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">42</span>, <span class="dv">51</span>, <span class="dv">59</span>, <span class="dv">64</span>, <span class="dv">76</span>, <span class="dv">93</span>, <span class="dv">106</span>, <span class="dv">125</span>, <span class="dv">149</span>, <span class="dv">171</span>, <span class="dv">199</span>, <span class="dv">205</span>)
chick_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">40</span>, <span class="dv">49</span>, <span class="dv">58</span>, <span class="dv">72</span>, <span class="dv">84</span>, <span class="dv">103</span>, <span class="dv">122</span>, <span class="dv">138</span>, <span class="dv">162</span>, <span class="dv">187</span>, <span class="dv">209</span>, <span class="dv">215</span>)
chick_<span class="dv">3</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">42</span>, <span class="dv">53</span>, <span class="dv">62</span>, <span class="dv">73</span>, <span class="dv">85</span>, <span class="dv">102</span>, <span class="dv">123</span>, <span class="dv">138</span>, <span class="dv">170</span>, <span class="dv">204</span>, <span class="dv">235</span>, <span class="dv">256</span>)
chick_<span class="dv">4</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">41</span>, <span class="dv">49</span>, <span class="dv">61</span>, <span class="dv">74</span>, <span class="dv">98</span>, <span class="dv">109</span>, <span class="dv">128</span>, <span class="dv">154</span>, <span class="dv">192</span>, <span class="dv">232</span>, <span class="dv">280</span>, <span class="dv">290</span>)</code></pre>
</div>
<div id="chicken-bar-plot" class="section level3">
<h3><span class="header-section-number">10.2.8</span> Chicken bar plot</h3>
<p>With the data from the previous question, create a bar plot of the maximum weights of the chicks.</p>
</div>
<div id="discoveries" class="section level3">
<h3><span class="header-section-number">10.2.9</span> Discoveries</h3>
<p>The R language comes with a wealth of datasets for you to use as practice materials. We will see several of these. One of these datasets is The Time-Series dataset called <code>discoveries</code> holding the numbers of “great” inventions and scientific discoveries in each year from 1860 to 1959. Type its name in the console to see it. Create plot(s) answering these questions:</p>
<p><strong>A</strong><br />
What is the frequency distribution of the number of discoveries per year? Use the <code>barplot()</code> and <code>table()</code> functions for this.</p>
<p><strong>B</strong><br />
What is the 5-number summary of discoveries per year?</p>
<p><strong>C</strong><br />
What is the trend over time for the numbers of discoveries per year?</p>
<p>PS: This is actually not a simple vector but a vector with some time-related attributes. It is called a Time-Series (a <code>ts</code> class), but this does not really matter for this assignment.</p>
</div>
<div id="lung-cancer" class="section level3">
<h3><span class="header-section-number">10.2.10</span> Lung cancer</h3>
<p>The R datasets package has three related timeseries datasets relating to lung cancer deaths. These are <code>ldeaths</code>, <code>mdeaths</code> and <code>fdeaths</code> for total, male and female deaths respectively. Create a line plot showing the monthly mortality holding all three of these datasets. Use the <code>legend()</code> function to add a legend to the plot, as demonstrated in this example:</p>
<pre class="sourceCode r"><code class="sourceCode r">t &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">5</span>
y1 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">6</span>)
y2 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">7</span>)
<span class="kw">plot</span>(t, y1, <span class="dt">type =</span> <span class="st">&quot;b&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;response&quot;</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">8</span>))
<span class="kw">points</span>(t, y2, <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="dt">type =</span> <span class="st">&quot;b&quot;</span>)
<span class="kw">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;series 1&quot;</span>, <span class="st">&quot;series 2&quot;</span>), <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="dt">pch =</span> <span class="dv">1</span>, <span class="dt">lty =</span> <span class="dv">1</span>)</code></pre>
<p><img src="davur_ebook_files/figure-html/legend-demo-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p><strong>A</strong><br />
Create the mentioned line plot. Do you see trends and/or patterns and if so, can you explain these?</p>
<p><strong>B</strong><br />
Create a combined boxplot of the three time-series. Are there outliers? If so, can you figure out when this occurred, and why?</p>
</div>
</div>
<div id="complex-datatypes" class="section level2">
<h2><span class="header-section-number">10.3</span> Complex datatypes</h2>
<p>This section serves you some datatype challenges.</p>
<div id="creating-factors" class="section level3">
<h3><span class="header-section-number">10.3.1</span> Creating factors</h3>
<p><strong>A</strong><br />
Given this vector:</p>
<pre class="sourceCode r"><code class="sourceCode r">animal_risk &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">1</span>)</code></pre>
<p>and these possible levels:
1: harmless
2: risky
3: dangerous
4: deadly</p>
<p>Create a factor from this data and then barplot the result.</p>
<p><strong>B</strong><br />
Given this data, a simulation of wealth distribution of “poor”, “middle class”, “wealthy” &quot;rich:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">1234</span>)
wealth_male &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dt">x =</span> letters[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>], 
                 <span class="dt">size =</span> <span class="dv">1000</span>,
                 <span class="dt">replace=</span> <span class="ot">TRUE</span>, 
                 <span class="dt">prob =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, <span class="fl">0.17</span>, <span class="fl">0.12</span>, <span class="fl">0.01</span>))
wealth_female &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dt">x =</span> letters[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>], 
                 <span class="dt">size =</span> <span class="dv">1000</span>,
                 <span class="dt">replace=</span> <span class="ot">TRUE</span>, 
                 <span class="dt">prob =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.15</span>, <span class="fl">0.497</span>, <span class="fl">0.003</span>))</code></pre>
<p>Create a factor from these two and report the cumulative percentage of its individual levels starting at the most abundant level, combined for male and female. Hint: use <code>table()</code> and <code>prop.table()</code>.</p>
<p>Next, create a side-by-side barplot of this data. Don’t forget the legend!</p>
</div>
<div id="a-dictionary-with-a-named-vector" class="section level3">
<h3><span class="header-section-number">10.3.2</span> A dictionary with a named vector</h3>
<p>Almost all programming languages know the (hash)map / dictionary data structure storing so-called “key-and-value” pairs. They make it possible to “look up” the value belonging to a “key”. That is where the term dictionary comes from. A dictionary holds keys (the words) and their meaning (values). R does not have a dictionary type but you could make a dict-like structure using a <strong><em>vector with named elements</em></strong>. Here follows an example.</p>
<p>If I wanted to create and use a DNA codon translation table, and use it to translate a piece of DNA, I could do something like what is shown below (there are only 4 of the 64 codons included). See if you can figure out what is going on there</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">## define codon table as named vector</span>
codons &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Gly&quot;</span>, <span class="st">&quot;Pro&quot;</span>, <span class="st">&quot;Lys&quot;</span>, <span class="st">&quot;Ser&quot;</span>)
<span class="kw">names</span>(codons) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GGA&quot;</span>, <span class="st">&quot;CCU&quot;</span>, <span class="st">&quot;AAA&quot;</span>, <span class="st">&quot;AGU&quot;</span>)

<span class="co">## the DNA to translate</span>
my_DNA &lt;-<span class="st"> &quot;GGACCUAAAAGU&quot;</span>
my_prot &lt;-<span class="st"> &quot;&quot;</span>
<span class="co">## iterate the DNA and take only every position</span>
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq</span>(<span class="dv">1</span>, <span class="kw">nchar</span>(my_DNA), <span class="dt">by=</span><span class="dv">3</span>)) {
    codon &lt;-<span class="st"> </span><span class="kw">substr</span>(my_DNA, i, i<span class="op">+</span><span class="dv">2</span>);
    my_prot &lt;-<span class="st"> </span><span class="kw">paste</span>(my_prot, codons[codon])
}
<span class="kw">print</span>(my_prot)</code></pre>
<pre><code>## [1] &quot; Gly Pro Lys Ser&quot;</code></pre>
<p><strong>A</strong><br />
Make a modified copy of this code chunk in such a way that no spaces are present between the amino acid residues (use help on <code>paste()</code> to figure this out) and that single-letter codes of amino acids are used instead of three-letter codes.</p>
<p><strong>B</strong><br />
<strong>[Challenge]</strong> Here is a vector called <code>nuc_weights</code>. It holds the weights for the nucleotides A, C, G and U respectively. Make it a named vector, iterate <code>my_DNA</code> from the above code chunk and calculate its molecular weight.</p>
<pre class="sourceCode r"><code class="sourceCode r">nuc_weights &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">491.2</span>, <span class="fl">467.2</span>, <span class="fl">507.2</span>, <span class="fl">482.2</span>)</code></pre>
</div>
<div id="protein-concentrations-with-lowry" class="section level3">
<h3><span class="header-section-number">10.3.3</span> Protein concentrations with Lowry</h3>
<p>The Lowry method is a widely used spectroscopic method to quantify protein amounts in solutions. Here are some real Lowry results for the quantification of proteins:
The calibration curve is made using BSA (bovine serum albumin) as a standard.
The concentrations are in mg/ml:</p>
<pre class="sourceCode r"><code class="sourceCode r">bsa_conc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>,    <span class="dv">0</span>,  <span class="fl">0.025</span>,  <span class="fl">0.025</span>,  <span class="fl">0.075</span>,  <span class="fl">0.075</span>,  <span class="fl">0.125</span>,  <span class="fl">0.125</span>)</code></pre>
<p>Note that the duplos are all put into one vector. We’ll deal with that later.</p>
<p>These are the obtained absorption values at 750 nm (again for the duplos):</p>
<pre class="sourceCode r"><code class="sourceCode r">OD750 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.063</span>,   <span class="fl">0.033</span>,  <span class="fl">0.16</span>,   <span class="fl">0.181</span>,  <span class="fl">0.346</span>,  <span class="fl">0.352</span>,  <span class="fl">0.491</span>,  <span class="fl">0.488</span>)</code></pre>
<p><strong>A</strong></p>
<p>Combine these two vectors in a dataframe and assign to a variable with name <code>dilution</code>. Do not add column names yet!</p>
<p><strong>B</strong></p>
<p>Now add the appropriate column names: <code>prot_conc</code> and <code>absorption</code>.</p>
<p><strong>C</strong></p>
<p>You forgot to include the last datapoints (again these are duplos):</p>
<pre class="sourceCode r"><code class="sourceCode r">bsa_conc2 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.175</span>,   <span class="fl">0.175</span>,  <span class="fl">0.25</span>,   <span class="fl">0.25</span>)
OD750_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.597</span>, <span class="fl">0.595</span>,  <span class="fl">0.743</span>,  <span class="fl">0.742</span>)</code></pre>
<p>Generate a dataframe assigned to the variable <code>df_temp</code> from these datapoints. This time, add the appropriate column names with creation of the dataframe. be sure you use exactly the same comlumn names.</p>
<p><strong>D</strong></p>
<p>Now add the second dataframe <code>df_temp</code> to dataframe with name <code>dilution</code> and assign it to the name <code>dilution</code> (thus overwrite the original variable):</p>
<p><strong>E</strong></p>
<p>Generate a line plot with the following properties:</p>
<ul>
<li>Title: “Absorbance as a function of protein concentration”.<br />
</li>
<li>Axis labels with clear indicated quantities and units.<br />
</li>
<li>Datapoints and connector lines should be visible and have a blue color.<br />
</li>
<li>Y-axis lower limit should be 0 and the upper limit should be 1.<br />
</li>
<li>The plot character (symbol) should be a filled circle.</li>
</ul>
<p>Use <code>col = rgb(0, 0, 1, 0.5)</code> to get transparent blue plot symbols.</p>
<p><strong>F</strong></p>
<p>Generate a new dataframe from the dataframe <code>dilution</code>, now with the duplo measurements side-by-side. So, instead of 2 you will now have 4 columns.</p>
<ul>
<li>Fist generate a dataframe <code>even</code> using the even numbered rows.<br />
</li>
<li>Then generate a dataframe <code>odd</code>using the odd numbered rows.<br />
</li>
<li>Finally combine these two new dataframes into one and assign it to the variable <code>dilution_duplo</code>.</li>
</ul>
<p>Hint: use subsetting with TRUE and FALSE and vector cycling.</p>
<p><strong>G</strong></p>
<p>The result will be a dataframe with two columns named <code>prot_conc</code> and two columns named <code>absorption</code>. Delete the second column named <code>prot_conc</code> (this is an exact copy after all!) and rename the column names to <code>prot_conc</code>, <code>abs1</code> and <code>abs2</code>.</p>
<p><strong>H</strong></p>
<p>Calculate the mean of the two abs measurements and add it as a column named <code>mean_abs</code>:</p>
<p><strong>I</strong></p>
<p>Generate a <strong>bar</strong> plot with the following properties:</p>
<ul>
<li>Title: “Absorbance versus protein concentration”.<br />
</li>
<li>Axis labels with clear indicated quantities and units.<br />
</li>
<li>Y-axis lower-limit should be 0, upper-limit should be 1.<br />
</li>
<li>Plot the mean absorbance on the Y-axis.<br />
</li>
<li>Bars should have a green color.</li>
</ul>
</div>
<div id="hplc-data" class="section level3">
<h3><span class="header-section-number">10.3.4</span> HPLC data</h3>
<p>The data below are from an educational experiment “Thymol quantification in Thyme”.
This data are obtained by a reverse-phase HPLC method using thymol in a standard curve. Peak areas were calculated using an integrator device. Note that peak areas are dimensionless. The concentration thymol is in μg/ml.</p>
<ul>
<li>co = concentration</li>
<li>pa = peak area</li>
</ul>
<pre class="sourceCode r"><code class="sourceCode r">co &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">100</span>,    <span class="dv">100</span>,    <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">5</span>, <span class="dv">5</span>)
pa &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1969017</span>,    <span class="dv">1858012</span>,    <span class="dv">1399762</span>,    <span class="dv">1449423</span>,    <span class="dv">963014</span>, <span class="dv">832137</span>, <span class="dv">467856</span>, <span class="dv">562012</span>, <span class="dv">200123</span>, <span class="dv">145545</span>, <span class="dv">94567</span>,  <span class="dv">64752</span>)</code></pre>
<p><strong>A</strong></p>
<p>Add the vectors to a dataframe named <code>hplc_data</code>. Use appropriate and clear column names.</p>
<p><strong>B</strong></p>
<p>The concentrations are in descending order. Use <code>order()</code> to sort the dataframe so that the concentrations are in ascending order.</p>
<p><strong>C</strong></p>
<p>Generate a scatter plot with the following settings:
- Title: “Peak area as a function of thymol concentration”
- Axis labels with clear indicated quantities and units.
- Datapoints should be visible (no connector lines).
- Datapoints should be blue
- Add a red colored linear regression line with a lineweigth of 2.</p>
</div>
<div id="airquality" class="section level3">
<h3><span class="header-section-number">10.3.5</span> Airquality</h3>
<p>The <code>airquality</code> dataset is also one of the datasets included in the <code>datasets</code> package. We’ll explore this for a few questions.</p>
<p><strong>A</strong><br />
Create a scatterplot of Temperature as a function of Solar radiation. Is there, as you might naively expect, a strong correlation? You could use <code>cor.test()</code> to find out. Add a linear model using <code>lm()</code> to extend your plot.</p>
<p><strong>B</strong><br />
Create a boxplot-series of <code>Temp</code> as a function of <code>Month</code> (use <code>?boxplot</code> to find out how this works). What appears to be the warmest month?</p>
<p><strong>C</strong><br />
What date (day/month) has the lowest recorded temperature? Which the highest? Please give temperature values in Celsius, not Fahrenheit! (Yes, this is an extra challenge!)</p>
<p><strong>D</strong><br />
Create a histogram of the wind speeds, and add a thick blue vertical line for the value of the mean and a fat red line for the median (use <code>abline()</code> for this).</p>
<p><strong>E</strong><br />
Use the <code>pairs()</code> function with argument <code>panel = panel.smooth</code> to plot all pairwise correlations between Ozone, Solar radiation, Wind and Temperature. Which pair shows the strongest correlation in your opinion? Verify this using the <code>cor()</code> function after removing incomplete cases. Create a separate well annotated scatterplot of this pair.</p>
<p>[<strong>Challenge</strong>] Through the <code>summary()</code> function you can obtain the R-squared value -the strength of the correlation- like this: <code>summary(your_linear_model)$r.squared</code>. Can you place it within the plot using the <code>text()</code> function?</p>
</div>
<div id="file-reading-practice" class="section level3">
<h3><span class="header-section-number">10.3.6</span> File reading practice</h3>
<p>The files in the <a href="data/file_reading/">data/file_reading/</a> directory contain some (simulated) gene-array data. The dataset contains a selection of induced transcripts after some stimulus. The columns represent:</p>
<ul>
<li>the mRNA entry<br />
</li>
<li>fold induction after some stimulus<br />
</li>
<li>the protein entry number<br />
</li>
<li>protein length in number of amino acids of the corresponding NP entry<br />
</li>
<li>the protein family the protein belongs to<br />
</li>
<li>the predicted cellular localization</li>
</ul>
<p>Whatever the contents of a file, you always need to address (some of) these questions:</p>
<ul>
<li>Are there comment lines at the top?<br />
</li>
<li>Is there a header line with column names?<br />
</li>
<li>What is the column separator?<br />
</li>
<li>Are there quotes around character data?<br />
</li>
<li>How are missing values encoded?<br />
</li>
<li>How are numeric values encoded?<br />
</li>
<li>What is the type in each column?</li>
</ul>
<p>Also: read the help for the <code>read.table</code> function carefully.</p>
<p>Read the content of the each file in the <code>file_reading</code> directory into a dataframe object and assign it to the variable name <code>df</code>. There are 15 different files to practice your file loading skills.</p>
</div>
<div id="bird-observations" class="section level3">
<h3><span class="header-section-number">10.3.7</span> Bird observations</h3>
<p>You will explore a bird observation dataset, downloaded from <a href="http://goldengateaudubon.org/birding-resources/observations/">GOLDEN GATE AUDUBON SOCIETY</a>. This file lists bird observations collected by this bird monitoring group in the San Francisco Bay Area. I already cleaned it a bit and placed it here: <a href="data/Observations-Data-2014.csv">data/Observations-Data-2014.csv</a>.</p>
<p>You can download it as follows:</p>
<pre class="sourceCode r"><code class="sourceCode r">file_name &lt;-<span class="st"> &quot;Observations-Data-2014.csv&quot;</span>
remote_url &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;https://raw.githubusercontent.com/MichielNoback/davur1_gitbook/master/data/&quot;</span>, file_name)

<span class="kw">download.file</span>(<span class="dt">url =</span> remote_url, <span class="dt">destfile =</span> file_name)</code></pre>
<p>Load the observation data into R and assign it to a variable called <code>bird_obs</code>.</p>
<p>From here on, it is assumed that you have the dataframe <code>bird_obs</code> loaded. This series of exercises deals with cleaning and transforming data, and exploring a cleaned dataset using basic plotting techniques and descriptive statistics.</p>
<p><strong>A</strong><br />
First, explore the raw data as they are.</p>
<ul>
<li>What data on bird observations were recorded (i.e. what kind of variables do you have)?</li>
<li>What did R do to the original column names?</li>
<li>Are all column names clear to you?</li>
</ul>
<p><strong>B</strong><br />
How many bird observations were recorded?</p>
<p><strong>C</strong><br />
The column holding observation “Number” is actually not a number. Into what type has R converted it?</p>
<p><strong>D</strong><br />
Convert the “Number” column into an integer column using <code>as.integer()</code>, but assign it to a new column called “Count” (i.e. do not overwrite the original values). Compare the first 50 values or so of these two columns. What happened to the data? Is this OK?</p>
<p><strong>E</strong><br />
The previous question has shown that converting factors to numbers is a bit dangerous. It is often easiest to convert characters to numbers. The best way to do this is by using the <code>as.is = c(&lt;column indices&gt;)</code> argument for the <code>read.table()</code> function.</p>
<p>So, which columns should be loaded as real factor data and which as plain character data? Use <code>read.table()</code> and the <code>as.is</code> argument to reload the data, and then transform the <code>Number</code> column to integer again as <code>Count</code>.</p>
<p><strong>F</strong><br />
Compare the first 50 values of the Number and Count columns again. Has the conversion succeeded? How many <code>Number</code> values could not be transformed into an integer value? Hint: use <code>is.na()</code></p>
<p><strong>G</strong><br />
Explore the sighting counts:</p>
<ul>
<li>Report the sighting with the maximum number of birds in a single sighting?</li>
<li>What is the mean sighting count?</li>
<li>What is the median of the sighting count?</li>
</ul>
<p><strong>H</strong><br />
Is the <code>Count</code> variable a normal distributed value? You can use <code>hist(...)</code>, <code>table()</code> or <code>plot(density(...))</code> to explore this further.</p>
<p><strong>I</strong><br />
Explore the species constitution:</p>
<ul>
<li>How many different species were recorded?</li>
<li>How many genera do they constitute?</li>
<li>What species from the genus “Puffinus” have been observed?</li>
</ul>
<p>Hint: use the function <code>unique()</code> here.</p>
<p>[<strong>Challenge</strong>] The number of unique “common names” should equal the number of unique “scientific names” since each species has a common name and a scientific name. The scientific name can be generated from Genus and Species. Investigate this. Is it correct? If not, what went wrong?</p>
<p><strong>J [Challenge]</strong><br />
This is a challenge exercise for those who like to grind their brains! Think of a strategy to “rescue” the NAs that appear after transforming “Number” to “Count”. Hint: use <code>gsub()</code> or<code>grep()</code></p>
</div>
</div>
<div id="regular-expressions" class="section level2">
<h2><span class="header-section-number">10.4</span> Regular Expressions</h2>
<div id="restriction-enzymes-1" class="section level3">
<h3><span class="header-section-number">10.4.1</span> Restriction enzymes</h3>
<p><strong>A</strong><br />
The restriction enzyme PacI has the recognition sequence “TTAATTAA”. Define (at least) three alternative regex patterns that will catch these sites.</p>
<p><strong>B</strong><br />
The restriction enzyme SfiI has the recognition sequence “GGCCNNNNNGGCC”. Define (at least) three alternative regex patterns that will catch these sites.</p>
</div>
<div id="prosite-patterns-1" class="section level3">
<h3><span class="header-section-number">10.4.2</span> Prosite Patterns</h3>
<p><strong>A</strong><br />
The Prosite pattern PS00211 (ABC-transporter-1; <a href="https://prosite.expasy.org/PS00211" class="uri">https://prosite.expasy.org/PS00211</a>) has the pattern:<br />
“[LIVMFYC]-[SA]-[SAPGLVFYKQH]-G-[DENQMW]-[KRQASPCLIMFW]-[KRNQSTAVM]-[KRACLVM]-[LIVMFYPAN]-{PHY}-[LIVMFW]-[SAGCLIVP]-{FYWHP}-{KRHP}-[LIVMFYWSTA].”
Translate it into a regex pattern. Info on the syntax is here: <a href="https://prosite.expasy.org/prosuser.html#conv_pa" class="uri">https://prosite.expasy.org/prosuser.html#conv_pa</a></p>
<p><strong>B</strong><br />
The Prosite pattern PS00018 (EF-hand calcium-binding domain; <a href="https://prosite.expasy.org/PS00018" class="uri">https://prosite.expasy.org/PS00018</a>) has the pattern:
“D-{W}-[DNS]-{ILVFYW}-[DENSTG]-[DNQGHRK]-{GP}-[LIVMC]-[DENQSTAGC]-x(2)- [DE]-[LIVMFYW].”
Translate it into a regex pattern.</p>
<p>You could exercise more by simply browsing Prosite. Test your pattern by fetching the proteins referred to within the Prosite pattern details page.</p>
</div>
<div id="fasta-headers" class="section level3">
<h3><span class="header-section-number">10.4.3</span> Fasta Headers</h3>
<p>The fasta sequence format is a very common sequence file format used in molecular biology.
It looks like this (I omitted most of the actual protein sequences for better representation):</p>
<pre><code>&gt;gi|21595364|gb|AAH32336.1| FHIT protein [Homo sapiens]
MSFRFGQHLIK...ALRVYFQ
&gt;gi|15215093|gb|AAH12662.1| Fhit protein [Mus musculus]
MSFRFGQHLIK...RVYFQA
&gt;gi|151554847|gb|AAI47994.1| FHIT protein [Bos taurus]
MSFRFGQHLIK...LRVYFQ</code></pre>
<p>As you can see there are several distinct elements within the Fasta <strong><em>header</em></strong> which is the description line above the actual sequence: one or more database identification strings, a protein description or name and an organism name. Study the format - we are going to extract some elements from these fasta headers using the <code>stringr</code> package. Install it if you don’t have it yet.</p>
<p>Here is a small example:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(stringr)
hinfII_re &lt;-<span class="st"> &quot;GA[GATC]TC&quot;</span>
sequences &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GGGAATCC&quot;</span>, <span class="st">&quot;TCGATTCGC&quot;</span>, <span class="st">&quot;ACGAGTCTA&quot;</span>)
<span class="kw">str_extract</span>(<span class="dt">string =</span> sequences,
            <span class="dt">pattern =</span> hinfII_re)</code></pre>
<pre><code>## [1] &quot;GAATC&quot; &quot;GATTC&quot; &quot;GAGTC&quot;</code></pre>
<p>Function <code>str_extract()</code> simply extracts the exact match of your regex (shown above). On the other hand, function <code>str_match()</code> supports <strong><em>grouping capture</em></strong> through bounding parentheses:</p>
<pre class="sourceCode r"><code class="sourceCode r">phones &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;+31-6-23415239&quot;</span>, <span class="st">&quot;+49-51-55523146&quot;</span>, <span class="st">&quot;+31-50-5956566&quot;</span>)
phones_re &lt;-<span class="st"> &quot;</span><span class="ch">\\</span><span class="st">+(</span><span class="ch">\\</span><span class="st">d{2})-(</span><span class="ch">\\</span><span class="st">d{1,2})&quot;</span> <span class="co">#matching country codes and area codes</span>
matches &lt;-<span class="st"> </span><span class="kw">str_match</span>(phones, phones_re) 
matches</code></pre>
<pre><code>##      [,1]     [,2] [,3]
## [1,] &quot;+31-6&quot;  &quot;31&quot; &quot;6&quot; 
## [2,] &quot;+49-51&quot; &quot;49&quot; &quot;51&quot;
## [3,] &quot;+31-50&quot; &quot;31&quot; &quot;50&quot;</code></pre>
<p>Thus, each set of parentheses will yield a column in the returned matrix. Simply use its column index to get that result set:</p>
<pre class="sourceCode r"><code class="sourceCode r">matches[, <span class="dv">2</span>] <span class="co">##the country codes</span></code></pre>
<pre><code>## [1] &quot;31&quot; &quot;49&quot; &quot;31&quot;</code></pre>
<p>Now, given the fasta headers in <a href="./data/fasta_headers.txt">./data/fasta_headers.txt</a>
which you can simply load into a character vector using <code>readLines()</code>, extract the following.</p>
<p><strong>A</strong><br />
- Extract all complete organism names.<br />
- Extract all species-level organism names (omitting subspecies and strains etc).</p>
<p><strong>B</strong><br />
Extract all <strong><em>first</em></strong> database identifiers. So in this header element <code>&gt;gi|224017144|gb|EEF75156.1|</code> you should extract only <code>gi|224017144</code></p>
<p><strong>C</strong><br />
Extract all protein names/descriptions.</p>
</div>
</div>
<div id="scripting-2" class="section level2">
<h2><span class="header-section-number">10.5</span> Scripting</h2>
<p>This section serves you some exercises that will help you improve your function-writing skills.</p>
<div id="illegal-reproductions" class="section level3">
<h3><span class="header-section-number">10.5.1</span> Illegal reproductions</h3>
<p>As an exercise, you will re-invent the wheel here for some statistical functions.</p>
<div id="the-mean" class="section level4 unnumbered">
<h4>The mean</h4>
<p>Create a function, <code>my_mean()</code>, that duplicates the R function <code>mean()</code>, i.e. calculates and returns the mean of a vector of numbers, without actually using <code>mean()</code>.</p>
</div>
<div id="standard-deviation" class="section level4 unnumbered">
<h4>Standard deviation</h4>
<p>Create a function, <code>my_sd()</code>, that duplicates the R function <code>sd()</code>, i.e. calculates and returns the standard deviation of a vector of numbers, without actually using <code>sd()</code>.</p>
</div>
<div id="median" class="section level4 unnumbered">
<h4>Median</h4>
<p><strong>[Challenge]</strong> Create a function, <code>my_median()</code>, that duplicates the R function <code>median()</code>, i.e. calculates and returns the median of a vector of numbers. This is actually a bit harder than you might expect. Hint: use the <code>sort()</code> function.</p>
</div>
</div>
<div id="interquantile-ranges" class="section level3">
<h3><span class="header-section-number">10.5.2</span> Interquantile ranges</h3>
<p>Create a function that will calculate a custom “interquantile range”. The function should accept three arguments: a numeric vector, a lower quantile and an upper quantile. It should return the difference (range) between these two quantile values. The lower quantile should default to 0 and the higher to 1, thus returning <code>max(x)</code> minus <code>min(x)</code>. The function therefore has this “signature”:</p>
<pre class="sourceCode r"><code class="sourceCode r">interquantile_range &lt;-<span class="st"> </span><span class="cf">function</span>(x, <span class="dt">lower =</span> <span class="dv">0</span>, <span class="dt">higher =</span> <span class="dv">100</span>) {}</code></pre>
<p>Perform some tests on the arguments to make a robust method: are all arguments numeric?</p>
<p>To test you method, you can compare <code>interquantile_range(some_vector, 0.25, 0.75)</code> with <code>IQR(some_vector)</code> - they should be the same.</p>
</div>
<div id="vector-distance" class="section level3">
<h3><span class="header-section-number">10.5.3</span> Vector distance</h3>
<p>Create a function, <code>distance(p, q)</code>, that will calculate and return the Euclidean distance between two vectors of equal length. A numeric vector can be seen as a point in multidimensional space. Euclidean distance is defined as</p>
<p><span class="math display">\[d(p, q) = \sqrt{\sum_{i = 1}^{n}(q_i-p_i)^2}\]</span>
Where <em>p</em> and <em>q</em> are the two vectors and <em>n</em> the length of the two vectors.<br />
You should first perform a check whether the two vectors are of equal length and both of type <code>numeric</code> or <code>integer</code>. If not, the function should abort with an appropriate error message.</p>
<div id="other-distance-measures" class="section level4 unnumbered">
<h4>Other distance measures</h4>
<p>Extend the function of the previous assignment in such a way that a third argument is accepted, <code>method =</code>, which defaults to “euclidean”. Other possible distance measures are “Manhattan” (same as “city block” and “taxicab”) and [<strong>Challenge</strong>] Pearson correlation. Look the equations for these up in Wikipedia or some other place.</p>
</div>
</div>
<div id="gc-percentage-of-dna" class="section level3">
<h3><span class="header-section-number">10.5.4</span> G/C percentage of DNA</h3>
<p><strong>[Challenge XL]</strong> Create a function, <code>GC_perc()</code>, that calculates and returns the GC percentage of a DNA or RNA sequence. Accept as input a sequence and a flag -<code>strict</code>- indicating whether other characters are accepted than core DNA/RNA (GATUC). If <code>strict = FALSE</code>, the percentage of other characters should be reported using a <code>warning()</code> call. If <code>strict = TRUE</code>, the function should terminate with an error message. Use <code>stop()</code> for this. <code>strict</code> should default to <code>TRUE</code>. NOTE, usage of <code>strict</code> can complicate things, so start with the core functionality!
You can use <code>strsplit()</code> or <code>substr()</code> to get hold of individual sequence characters.</p>
</div>
</div>
<div id="function-apply-and-its-relatives" class="section level2">
<h2><span class="header-section-number">10.6</span> Function <code>apply</code> and its relatives</h2>
<p>In this section you will encounter some exercises revolving around the different flavors of apply.</p>
<div id="whale-selenium" class="section level3">
<h3><span class="header-section-number">10.6.1</span> Whale selenium</h3>
<p>On the course website under <a href="https://michielnoback.github.io/bincourses/course_contents/davur/resources.html">Resources</a> you will find a link to file <code>whale_selenium.txt</code>. You could download it into your working directory manually or use <code>download.file()</code> to obtain it. However, there is a third way to get its contents without actually downloading it as a local copy. You can read it directly using <code>read.table()</code> as shown here.</p>
<pre class="sourceCode r"><code class="sourceCode r">whale_sel_url &lt;-<span class="st"> &quot;https://raw.githubusercontent.com/MichielNoback/davur1/gh-pages/exercises/data/whale_selenium.txt&quot;</span>
whale_selenium &lt;-<span class="st"> </span><span class="kw">read.table</span>(whale_sel_url,
    <span class="dt">header =</span> T,
    <span class="dt">row.names =</span> <span class="dv">1</span>)</code></pre>
<p>Note: when you are going to load a file many times it is probably better to store a local copy.</p>
<p><strong>A</strong><br />
Report the means of both columns using <code>apply()</code>.</p>
<p><strong>B</strong>
Report the standard deviation of both columns, using <code>apply()</code></p>
<p><strong>C</strong><br />
Report the standard error of the mean of both columns, using <code>apply()</code> The SEM is calculated as <span class="math display">\[\frac{sd}{\sqrt{n}}\]</span> where <span class="math inline">\(sd\)</span> is the sample standard deviation and <span class="math inline">\(n\)</span> the number of measurements. You should create the function calculating this statistic yourself.</p>
<p><strong>D</strong><br />
Using <code>apply()</code>, calculate the ratio of <span class="math inline">\(Se_{tooth} / Se_{liver}\)</span> and attach it to the <code>whale_selenium</code> dataframe as column <code>ratio</code>. Create a histogram of this ratio.</p>
<p><strong>E</strong><br />
Using <code>print()</code> and <code>paste()</code>, report the mean and the standard deviation of the ratio column, but do this with an inline expression, e.g. an expression embedded in the R markdown paragraph text.</p>
</div>
<div id="chickweight" class="section level3">
<h3><span class="header-section-number">10.6.2</span> ChickWeight</h3>
<p>This exercise revolves around the <code>ChickWeight</code> dataset of the built-in <code>datasets</code> package.</p>
<p><strong>A</strong><br />
Report the number of chickens used in the experiment.</p>
<p><strong>B</strong><br />
Use <code>aggregate()</code> to get the mean weight of the chickens for the different Diets.</p>
<p><strong>C</strong><br />
Use <code>coplot()</code> to plot a panel with weight as function of Time, split over Diet.</p>
<p><strong>D</strong><br />
[<strong>Challenge</strong>] Add a column called <code>weight_gain</code> to the dataframe holding values for the weight gain since the last measurement. Take special care with rows marking the boundaries between individual chickens! You could consider using a traditional for loop here.
In the next course, we’ll see a more efficient way of doing this.</p>
<p><strong>E</strong><br />
Split the <code>weight_gain</code> column on Diet and report the mean, median and standard deviation for each diet.
If you were not successful in the previous question, load and attach the data from file <code>ChickWeight_weight_gain.Rdata</code> downloadable from <a href="https://github.com/MichielNoback/davur1_gitbook/raw/master/data/ChickWeight_weight_gain.Rdata">https://github.com/MichielNoback/davur1_gitbook/raw/master/data/ChickWeight_weight_gain.Rdata</a>. You can use this code chunk for downloading and loading the data into variable <code>stored_weight_gain</code>. Don’t forget to attach the column to the data frame!</p>
<pre class="sourceCode r"><code class="sourceCode r">local_file &lt;-<span class="st"> &quot;ChickWeight_weight_gain.Rdata&quot;</span>
<span class="kw">download.file</span>(<span class="kw">paste0</span>(<span class="st">&quot;https://github.com/MichielNoback/davur1_gitbook/raw/master/data/&quot;</span>, local_file), local_file)
<span class="kw">load</span>(local_file)</code></pre>
<p><strong>F</strong><br />
Create a (single-panel) boxplot for weight gain, split over Diet. Hint: read the <code>boxplot()</code> help page!</p>
</div>
<div id="food-constituents" class="section level3">
<h3><span class="header-section-number">10.6.3</span> Food constituents</h3>
<p>The <a href="https://raw.githubusercontent.com/MichielNoback/davur1_gitbook/master/data/food_constituents.txt">food constituents dataset</a> holds information on ingredients for different foods. Individual foods are simply marked with an id.</p>
<p><strong>A</strong><br />
Load the data and report the different food categories (<code>Type</code>). Also report the numbers of entries for each Type.</p>
<p><strong>B</strong><br />
What is the mean energy content of chocolate foods?</p>
<p><strong>C</strong><br />
What is the food category with the highest mean fat content?</p>
<p><strong>D</strong><br />
What food category has the highest mean energy content, and which has the lowest?</p>
<p><strong>E</strong><br />
<strong>[Challenge]</strong> Create a boxplot showing the difference in sugar content between drink and solid food.</p>
<p><strong>F</strong><br />
Assuming both saturated fats and sugar are bad for you, what food category do you consider the worst? Think of a means to answer this, explain it and carry it out.</p>
</div>
<div id="urine-properties" class="section level3">
<h3><span class="header-section-number">10.6.4</span> Urine properties</h3>
<p>The <a href="https://github.com/MichielNoback/datasets/tree/master/urine">urine specimens dataset</a> contains a <code>readme.txt</code> file and a data file (<code>urine.csv</code>; direct link: “<a href="https://raw.githubusercontent.com/MichielNoback/datasets/master/urine/urine.csv" class="uri">https://raw.githubusercontent.com/MichielNoback/datasets/master/urine/urine.csv</a>”). Study the readme to get an idea of the data. Download the file like this:</p>
<pre class="sourceCode r"><code class="sourceCode r">urine_file_name &lt;-<span class="st"> &quot;urine.csv&quot;</span>
url &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;https://raw.githubusercontent.com/MichielNoback/datasets/master/urine/&quot;</span>, urine_file_name)
local_name &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;../&quot;</span>, urine_file_name) <span class="co">#specifiy your own folder!</span>
<span class="kw">download.file</span>(<span class="dt">url =</span> url, <span class="dt">destfile =</span> local_name)</code></pre>
<p><strong>A</strong><br />
Load the data into a dataframe with name <code>urine</code>.</p>
<p><strong>B</strong><br />
Convert the column <code>r</code> into a factor with two levels: <code>yes</code> and <code>no</code>, and give it a better name: <code>ox_crystals</code>.</p>
<p><strong>C</strong><br />
Using <code>apply()</code>, report the mean and standard deviation of the numeric columns only. Give these with only two decimal digits. Use a <em>named vector</em> so you get this output:</p>
<pre><code>     gravity   ph   osmo  cond   urea calc
mean    1.02 6.03 615.04 20.90 266.41 4.14
sd      0.01 0.72 238.25  7.95 131.25 3.26</code></pre>
<p><strong>D</strong><br />
Using <code>aggregate</code>, report the mean of the numeric columns, but split over the <code>ox_crystals</code> variable levels. Which variables seem most likely candidates for a relation with oxalate crystal formation.</p>
<p><strong>E</strong><br />
Use the <code>pairs()</code> plotting function to explore the pairwise relationships between the different numeric variables</p>
<p><strong>F</strong><br />
Use the <code>heatmap.2()</code> and <code>cor()</code> functions together to create a heatmap of pairwise variable correlations. You will need to install package <code>gplots</code> for the heatmap.2 function. Alternatively, use <code>heatmap()</code> from base R.</p>
<p>Which visualization do you prefer - the <code>pairs()</code> or the <code>heatmap.2()</code>?</p>
<p><strong>G</strong><br />
[<strong>Challenge</strong>] There does not seem to be an interesting correlation between pH and any of the other variables. Sometimes this is becuase you are not looking in enough detail. Let’s dig a little further. Use <code>cut()</code> to split the pH variable into a factor with three levels: “acidic”, “neutral” and “basic” with breakpoints between them at pH 5.5 and 7. Next, use the <code>split()</code> and <code>lapply()</code> functions to calculate the mean, meadian and sd of the other numeric variables for each level of this pH factor. This exercise can of course be done in many ways. one of them is by using an <code>apply()</code> <em>within</em> the applied function of <code>lapply()</code>.</p>
</div>
<div id="bird-observations-revisited" class="section level3">
<h3><span class="header-section-number">10.6.5</span> Bird observations revisited</h3>
<p>This exercise revisits the bird observations dataset. You can download it <a href="data/Observations-Data-2014.csv">here</a>. (Re)load the dataset.</p>
<p><strong>A</strong><br />
Report the number of observations per <code>County</code>. Use both a textual as a barplot representation. With the barplot, you should order the bars according to observation numbers.</p>
<p><strong>B</strong><br />
Report the number of observations per <code>Observer.1</code> but only for observers with more than 10 observations, ordered from high to low observation count. Use <code>order()</code> to achieve this.</p>
<p><strong>C</strong><br />
Which observer has the highest number of observations listed (and how many is that)?</p>
<p><strong>D</strong><br />
Report the different observed species (using <code>Common.name</code>) for each genus. <strong>[Challenge]</strong> Report only the 5 Genera with the highest number of observed species.</p>
<p><strong>E</strong><br />
<strong>[Challenge]</strong> Create a Dataframe holding the number of birds per day (use Date.start) and plot it with date on the x-axis and number of birds on the y-axis. Hint: use <code>as.Date()</code> to convert the character date to a real date field. See this page how you can do that <a href="http://www.statmethods.net/input/dates.html">Date Values</a>.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="exploratory-data-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="exercise-solutions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["davur_ebook.pdf", "davur_ebook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
