<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Exploratory Data Analysis | Data Analysis &amp; Visualization using R (1)</title>
  <meta name="description" content="This gitbook accompanies the course Data Analysis &amp; Visualization using R 1 (DAVuR1) of the minor Bioinformatics for Life Science Students. It is organized by the Hanze University of Applied Science." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Exploratory Data Analysis | Data Analysis &amp; Visualization using R (1)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This gitbook accompanies the course Data Analysis &amp; Visualization using R 1 (DAVuR1) of the minor Bioinformatics for Life Science Students. It is organized by the Hanze University of Applied Science." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Exploratory Data Analysis | Data Analysis &amp; Visualization using R (1)" />
  
  <meta name="twitter:description" content="This gitbook accompanies the course Data Analysis &amp; Visualization using R 1 (DAVuR1) of the minor Bioinformatics for Life Science Students. It is organized by the Hanze University of Applied Science." />
  

<meta name="author" content="Michiel Noback" />


<meta name="date" content="2020-10-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dataframe-manipulations-1.html"/>
<link rel="next" href="exercises.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DAVuR eBook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Getting started</a></li>
<li class="chapter" data-level="2" data-path="toolbox.html"><a href="toolbox.html"><i class="fa fa-check"></i><b>2</b> The toolbox</a><ul>
<li class="chapter" data-level="2.1" data-path="toolbox.html"><a href="toolbox.html#embarking-on-data-science"><i class="fa fa-check"></i><b>2.1</b> Embarking on Data Science</a></li>
<li class="chapter" data-level="2.2" data-path="toolbox.html"><a href="toolbox.html#why-do-statistical-programming"><i class="fa fa-check"></i><b>2.2</b> Why do statistical programming?</a><ul>
<li class="chapter" data-level="" data-path="toolbox.html"><a href="toolbox.html#overview-of-the-toolbox"><i class="fa fa-check"></i>Overview of the toolbox</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="toolbox.html"><a href="toolbox.html#tool-1-the-r-programming-language"><i class="fa fa-check"></i><b>2.3</b> Tool 1: The R programming language</a></li>
<li class="chapter" data-level="2.4" data-path="toolbox.html"><a href="toolbox.html#tool-2-rstudio-as-development-environment"><i class="fa fa-check"></i><b>2.4</b> Tool 2: RStudio as development environment</a></li>
<li class="chapter" data-level="2.5" data-path="toolbox.html"><a href="toolbox.html#tool-3-r-markdown"><i class="fa fa-check"></i><b>2.5</b> Tool 3: R Markdown</a></li>
<li class="chapter" data-level="2.6" data-path="toolbox.html"><a href="toolbox.html#resources"><i class="fa fa-check"></i><b>2.6</b> Resources</a><ul>
<li class="chapter" data-level="" data-path="toolbox.html"><a href="toolbox.html#swirl-course"><i class="fa fa-check"></i>Swirl course</a></li>
<li class="chapter" data-level="" data-path="toolbox.html"><a href="toolbox.html#data-files"><i class="fa fa-check"></i>Data Files</a></li>
<li class="chapter" data-level="" data-path="toolbox.html"><a href="toolbox.html#web-resources-and-references"><i class="fa fa-check"></i>Web resources and references</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-r-coding.html"><a href="basic-r-coding.html"><i class="fa fa-check"></i><b>3</b> Basic R - coding</a><ul>
<li class="chapter" data-level="3.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#first-look-at-vectors-fuctions-and-variables"><i class="fa fa-check"></i><b>3.1</b> First look at vectors, fuctions and variables</a><ul>
<li class="chapter" data-level="3.1.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#doing-math-in-the-console"><i class="fa fa-check"></i><b>3.1.1</b> Doing Math in the console</a></li>
<li class="chapter" data-level="3.1.2" data-path="basic-r-coding.html"><a href="basic-r-coding.html#an-expression-dissected"><i class="fa fa-check"></i><b>3.1.2</b> An expression dissected</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="basic-r-coding.html"><a href="basic-r-coding.html#functions"><i class="fa fa-check"></i><b>3.2</b> Functions</a><ul>
<li class="chapter" data-level="3.2.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#getting-help-on-a-function"><i class="fa fa-check"></i><b>3.2.1</b> Getting help on a function</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="basic-r-coding.html"><a href="basic-r-coding.html#variables"><i class="fa fa-check"></i><b>3.3</b> Variables</a></li>
<li class="chapter" data-level="3.4" data-path="basic-r-coding.html"><a href="basic-r-coding.html#vectors"><i class="fa fa-check"></i><b>3.4</b> Vectors</a><ul>
<li class="chapter" data-level="3.4.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#r-is-completely-vector-based"><i class="fa fa-check"></i><b>3.4.1</b> R is completely vector-based</a></li>
<li class="chapter" data-level="3.4.2" data-path="basic-r-coding.html"><a href="basic-r-coding.html#five-datatype-that-live-in-vectors"><i class="fa fa-check"></i><b>3.4.2</b> Five datatype that live in vectors</a></li>
<li class="chapter" data-level="3.4.3" data-path="basic-r-coding.html"><a href="basic-r-coding.html#creating-vectors"><i class="fa fa-check"></i><b>3.4.3</b> Creating vectors</a></li>
<li class="chapter" data-level="3.4.4" data-path="basic-r-coding.html"><a href="basic-r-coding.html#vector-fiddling"><i class="fa fa-check"></i><b>3.4.4</b> Vector fiddling</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="basic-r-coding.html"><a href="basic-r-coding.html#other-operators"><i class="fa fa-check"></i><b>3.5</b> Other operators</a><ul>
<li class="chapter" data-level="3.5.1" data-path="basic-r-coding.html"><a href="basic-r-coding.html#logical-operators"><i class="fa fa-check"></i><b>3.5.1</b> Logical operators</a></li>
<li class="chapter" data-level="3.5.2" data-path="basic-r-coding.html"><a href="basic-r-coding.html#modulo"><i class="fa fa-check"></i><b>3.5.2</b> Modulo: <code>%%</code></a></li>
<li class="chapter" data-level="3.5.3" data-path="basic-r-coding.html"><a href="basic-r-coding.html#integer-division-and-rounding"><i class="fa fa-check"></i><b>3.5.3</b> Integer division <code>%/%</code> and rounding</a></li>
<li class="chapter" data-level="3.5.4" data-path="basic-r-coding.html"><a href="basic-r-coding.html#the-in-operator"><i class="fa fa-check"></i><b>3.5.4</b> The <code>%in%</code> operator</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="basic-r-coding.html"><a href="basic-r-coding.html#vector-creation-methods"><i class="fa fa-check"></i><b>3.6</b> Vector creation methods</a></li>
<li class="chapter" data-level="3.7" data-path="basic-r-coding.html"><a href="basic-r-coding.html#selecting-vector-elements"><i class="fa fa-check"></i><b>3.7</b> Selecting vector elements</a></li>
<li class="chapter" data-level="3.8" data-path="basic-r-coding.html"><a href="basic-r-coding.html#some-coding-style-rules-rules-for-writing-code"><i class="fa fa-check"></i><b>3.8</b> Some coding style rules rules for writing code</a></li>
<li class="chapter" data-level="3.9" data-path="basic-r-coding.html"><a href="basic-r-coding.html#the-best-keyboard-shortcuts-for-rstudio"><i class="fa fa-check"></i><b>3.9</b> The best keyboard shortcuts for RStudio</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html"><i class="fa fa-check"></i><b>4</b> Basic R - plotting</a><ul>
<li class="chapter" data-level="4.1" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#basic-embedded-plot-types"><i class="fa fa-check"></i><b>4.1</b> Basic embedded plot types</a><ul>
<li class="chapter" data-level="4.1.1" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#scatter-and-line-plots"><i class="fa fa-check"></i><b>4.1.1</b> Scatter and line plots</a></li>
<li class="chapter" data-level="4.1.2" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#barplots"><i class="fa fa-check"></i><b>4.1.2</b> Barplots</a></li>
<li class="chapter" data-level="4.1.3" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#histograms"><i class="fa fa-check"></i><b>4.1.3</b> Histograms</a></li>
<li class="chapter" data-level="4.1.4" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#density-plot-as-alternative-to-hist"><i class="fa fa-check"></i><b>4.1.4</b> Density plot as alternative to <code>hist()</code></a></li>
<li class="chapter" data-level="4.1.5" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#boxplots"><i class="fa fa-check"></i><b>4.1.5</b> Boxplots</a></li>
<li class="chapter" data-level="4.1.6" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#adding-more-data-and-a-legend"><i class="fa fa-check"></i><b>4.1.6</b> Adding more data and a legend</a></li>
<li class="chapter" data-level="4.1.7" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#helper-lines-and-lm"><i class="fa fa-check"></i><b>4.1.7</b> Helper lines and <code>lm()</code></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="basic-r-plotting.html"><a href="basic-r-plotting.html#graphical-parameters-to-plot"><i class="fa fa-check"></i><b>4.2</b> Graphical parameters to <code>plot()</code></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html"><i class="fa fa-check"></i><b>5</b> Complex Datatypes and File Reading</a><ul>
<li class="chapter" data-level="5.1" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#matrices-are-vectors-with-dimensions"><i class="fa fa-check"></i><b>5.1</b> Matrices are vectors with dimensions</a></li>
<li class="chapter" data-level="5.2" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#factors-nominal-ordinal-scales"><i class="fa fa-check"></i><b>5.2</b> Factors: Nominal &amp; Ordinal scales</a></li>
<li class="chapter" data-level="5.3" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#lists"><i class="fa fa-check"></i><b>5.3</b> Lists</a></li>
<li class="chapter" data-level="5.4" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#dataframes"><i class="fa fa-check"></i><b>5.4</b> Dataframes</a><ul>
<li class="chapter" data-level="5.4.1" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#selections-on-dataframes"><i class="fa fa-check"></i><b>5.4.1</b> Selections on dataframes</a></li>
<li class="chapter" data-level="5.4.2" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#read-from-file-using-read.table"><i class="fa fa-check"></i><b>5.4.2</b> Read from file using <code>read.table()</code></a></li>
<li class="chapter" data-level="5.4.3" data-path="complex-datatypes-and-file-reading.html"><a href="complex-datatypes-and-file-reading.html#dataframe-manipulations"><i class="fa fa-check"></i><b>5.4.3</b> Dataframe manipulations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="functions-1.html"><a href="functions-1.html"><i class="fa fa-check"></i><b>6</b> Functions</a><ul>
<li class="chapter" data-level="6.1" data-path="functions-1.html"><a href="functions-1.html#dealing-with-nas"><i class="fa fa-check"></i><b>6.1</b> Dealing with NAs</a></li>
<li class="chapter" data-level="6.2" data-path="functions-1.html"><a href="functions-1.html#descriptive-statistics"><i class="fa fa-check"></i><b>6.2</b> Descriptive statistics</a></li>
<li class="chapter" data-level="6.3" data-path="functions-1.html"><a href="functions-1.html#general-purpose-functions"><i class="fa fa-check"></i><b>6.3</b> General purpose functions</a></li>
<li class="chapter" data-level="6.4" data-path="functions-1.html"><a href="functions-1.html#convert-numeric-vector-to-factor-cut"><i class="fa fa-check"></i><b>6.4</b> Convert numeric vector to factor: <code>cut()</code></a></li>
<li class="chapter" data-level="6.5" data-path="functions-1.html"><a href="functions-1.html#file-io-revisited"><i class="fa fa-check"></i><b>6.5</b> File I/O revisited</a></li>
<li class="chapter" data-level="6.6" data-path="functions-1.html"><a href="functions-1.html#pattern-matching"><i class="fa fa-check"></i><b>6.6</b> Pattern matching</a><ul>
<li class="chapter" data-level="6.6.1" data-path="functions-1.html"><a href="functions-1.html#regex-syntax"><i class="fa fa-check"></i><b>6.6.1</b> Regex syntax</a></li>
<li class="chapter" data-level="6.6.2" data-path="functions-1.html"><a href="functions-1.html#quantifiers"><i class="fa fa-check"></i><b>6.6.2</b> Quantifiers</a></li>
<li class="chapter" data-level="6.6.3" data-path="functions-1.html"><a href="functions-1.html#some-examples"><i class="fa fa-check"></i><b>6.6.3</b> Some examples</a></li>
<li class="chapter" data-level="6.6.4" data-path="functions-1.html"><a href="functions-1.html#anchoring"><i class="fa fa-check"></i><b>6.6.4</b> Anchoring</a></li>
<li class="chapter" data-level="6.6.5" data-path="functions-1.html"><a href="functions-1.html#metacharacters-special-character-classes"><i class="fa fa-check"></i><b>6.6.5</b> Metacharacters: Special character classes</a></li>
<li class="chapter" data-level="6.6.6" data-path="functions-1.html"><a href="functions-1.html#locating-and-extracting"><i class="fa fa-check"></i><b>6.6.6</b> Locating and Extracting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="scripting.html"><a href="scripting.html"><i class="fa fa-check"></i><b>7</b> Scripting</a><ul>
<li class="chapter" data-level="7.1" data-path="scripting.html"><a href="scripting.html#introduction"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="scripting.html"><a href="scripting.html#flow-control"><i class="fa fa-check"></i><b>7.2</b> Flow control</a><ul>
<li class="chapter" data-level="7.2.1" data-path="scripting.html"><a href="scripting.html#conditional-execution-with-ifelse"><i class="fa fa-check"></i><b>7.2.1</b> Conditional execution with <code>if/else</code></a></li>
<li class="chapter" data-level="7.2.2" data-path="scripting.html"><a href="scripting.html#iteration-with-for"><i class="fa fa-check"></i><b>7.2.2</b> Iteration with <code>for(){}</code></a></li>
<li class="chapter" data-level="7.2.3" data-path="scripting.html"><a href="scripting.html#conditional-iteration-with-while"><i class="fa fa-check"></i><b>7.2.3</b> Conditional iteration with <code>while(){}</code></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="scripting.html"><a href="scripting.html#creating-functions"><i class="fa fa-check"></i><b>7.3</b> Creating functions</a><ul>
<li class="chapter" data-level="7.3.1" data-path="scripting.html"><a href="scripting.html#errors-and-warnings"><i class="fa fa-check"></i><b>7.3.1</b> Errors and warnings</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="scripting.html"><a href="scripting.html#scripting-1"><i class="fa fa-check"></i><b>7.4</b> Scripting</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html"><i class="fa fa-check"></i><b>8</b> Dataframe manipulations</a><ul>
<li class="chapter" data-level="8.1" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#the-apply-family-of-functions"><i class="fa fa-check"></i><b>8.1</b> The <code>apply()</code> family of functions</a><ul>
<li class="chapter" data-level="8.1.1" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#apply-apply-functions-over-array-margins"><i class="fa fa-check"></i><b>8.1.1</b> <code>apply()</code>: Apply Functions Over Array Margins</a></li>
<li class="chapter" data-level="8.1.2" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#lapply-apply-a-function-over-a-list-or-vector"><i class="fa fa-check"></i><b>8.1.2</b> <code>lapply()</code>: Apply a Function over a List or Vector</a></li>
<li class="chapter" data-level="8.1.3" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#sapply-apply-a-function-over-a-list-or-vector-and-simplify"><i class="fa fa-check"></i><b>8.1.3</b> <code>sapply()</code>: Apply a Function over a List or Vector and Simplify</a></li>
<li class="chapter" data-level="8.1.4" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#tapply-apply-a-function-over-a-ragged-array"><i class="fa fa-check"></i><b>8.1.4</b> <code>tapply()</code>: Apply a Function Over a Ragged Array</a></li>
<li class="chapter" data-level="8.1.5" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#split-divide-into-groups-and-reassemble"><i class="fa fa-check"></i><b>8.1.5</b> <code>split()</code>: Divide into Groups and Reassemble</a></li>
<li class="chapter" data-level="8.1.6" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#aggregate-compute-summary-statistics-of-data-subsets"><i class="fa fa-check"></i><b>8.1.6</b> <code>aggregate()</code>: Compute Summary Statistics of Data Subsets</a></li>
<li class="chapter" data-level="8.1.7" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#many-roads-lead-to-rome"><i class="fa fa-check"></i><b>8.1.7</b> Many roads lead to Rome</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#usecases"><i class="fa fa-check"></i><b>8.2</b> Example Use Cases</a><ul>
<li class="chapter" data-level="8.2.1" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#dfselection"><i class="fa fa-check"></i><b>8.2.1</b> Dataframe Selections</a></li>
<li class="chapter" data-level="8.2.2" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#apply"><i class="fa fa-check"></i><b>8.2.2</b> Apply</a></li>
<li class="chapter" data-level="8.2.3" data-path="dataframe-manipulations-1.html"><a href="dataframe-manipulations-1.html#embeddeddf"><i class="fa fa-check"></i><b>8.2.3</b> Processing Embedded Dataframes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>9</b> Exploratory Data Analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#introduction-1"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#eda-of-the-yeast-dataset"><i class="fa fa-check"></i><b>9.2</b> EDA of the Yeast dataset</a><ul>
<li class="chapter" data-level="9.2.1" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#introduction-2"><i class="fa fa-check"></i><b>9.2.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2.2" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#data-loading-and-prepping"><i class="fa fa-check"></i><b>9.2.2</b> Data loading and prepping</a></li>
<li class="chapter" data-level="9.2.3" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#data-verification"><i class="fa fa-check"></i><b>9.2.3</b> Data verification</a></li>
<li class="chapter" data-level="9.2.4" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#attribute-summaries"><i class="fa fa-check"></i><b>9.2.4</b> Attribute summaries</a></li>
<li class="chapter" data-level="9.2.5" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#attribute-relationships"><i class="fa fa-check"></i><b>9.2.5</b> Attribute relationships</a></li>
<li class="chapter" data-level="9.2.6" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html#correlations-to-the-dependent-variable"><i class="fa fa-check"></i><b>9.2.6</b> Correlations to the dependent variable</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>10</b> Exercises</a><ul>
<li class="chapter" data-level="10.1" data-path="exercises.html"><a href="exercises.html#getting-started-1"><i class="fa fa-check"></i><b>10.1</b> Getting started</a><ul>
<li class="chapter" data-level="10.1.1" data-path="exercises.html"><a href="exercises.html#install-the-tools"><i class="fa fa-check"></i><b>10.1.1</b> Install the tools</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="exercises.html"><a href="exercises.html#toolbox-1"><i class="fa fa-check"></i><b>10.2</b> Toolbox</a><ul>
<li class="chapter" data-level="10.2.1" data-path="exercises.html"><a href="exercises.html#résumé"><i class="fa fa-check"></i><b>10.2.1</b> Résumé</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="exercises.html"><a href="exercises.html#basic-r"><i class="fa fa-check"></i><b>10.3</b> Basic R</a><ul>
<li class="chapter" data-level="10.3.1" data-path="exercises.html"><a href="exercises.html#math-in-the-console"><i class="fa fa-check"></i><b>10.3.1</b> Math in the console</a></li>
<li class="chapter" data-level="10.3.2" data-path="exercises.html"><a href="exercises.html#first-look-at-functions"><i class="fa fa-check"></i><b>10.3.2</b> First look at functions</a></li>
<li class="chapter" data-level="10.3.3" data-path="exercises.html"><a href="exercises.html#variables-1"><i class="fa fa-check"></i><b>10.3.3</b> Variables</a></li>
<li class="chapter" data-level="10.3.4" data-path="exercises.html"><a href="exercises.html#vectors-1"><i class="fa fa-check"></i><b>10.3.4</b> Vectors</a></li>
<li class="chapter" data-level="10.3.5" data-path="exercises.html"><a href="exercises.html#stair-walking-and-heart-rate"><i class="fa fa-check"></i><b>10.3.5</b> Stair walking and heart rate</a></li>
<li class="chapter" data-level="10.3.6" data-path="exercises.html"><a href="exercises.html#more-subjects"><i class="fa fa-check"></i><b>10.3.6</b> More subjects</a></li>
<li class="chapter" data-level="10.3.7" data-path="exercises.html"><a href="exercises.html#chickens-on-a-diet"><i class="fa fa-check"></i><b>10.3.7</b> Chickens on a diet</a></li>
<li class="chapter" data-level="10.3.8" data-path="exercises.html"><a href="exercises.html#chicken-bar-plot"><i class="fa fa-check"></i><b>10.3.8</b> Chicken bar plot</a></li>
<li class="chapter" data-level="10.3.9" data-path="exercises.html"><a href="exercises.html#discoveries"><i class="fa fa-check"></i><b>10.3.9</b> Discoveries</a></li>
<li class="chapter" data-level="10.3.10" data-path="exercises.html"><a href="exercises.html#lung-cancer"><i class="fa fa-check"></i><b>10.3.10</b> Lung cancer</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="exercises.html"><a href="exercises.html#complex-datatypes"><i class="fa fa-check"></i><b>10.4</b> Complex datatypes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="exercises.html"><a href="exercises.html#creating-factors"><i class="fa fa-check"></i><b>10.4.1</b> Creating factors</a></li>
<li class="chapter" data-level="10.4.2" data-path="exercises.html"><a href="exercises.html#a-dictionary-with-a-named-vector"><i class="fa fa-check"></i><b>10.4.2</b> A dictionary with a named vector</a></li>
<li class="chapter" data-level="10.4.3" data-path="exercises.html"><a href="exercises.html#protein-concentrations-with-lowry"><i class="fa fa-check"></i><b>10.4.3</b> Protein concentrations with Lowry</a></li>
<li class="chapter" data-level="10.4.4" data-path="exercises.html"><a href="exercises.html#hplc-data"><i class="fa fa-check"></i><b>10.4.4</b> HPLC data</a></li>
<li class="chapter" data-level="10.4.5" data-path="exercises.html"><a href="exercises.html#airquality"><i class="fa fa-check"></i><b>10.4.5</b> Airquality</a></li>
<li class="chapter" data-level="10.4.6" data-path="exercises.html"><a href="exercises.html#file-reading-practice"><i class="fa fa-check"></i><b>10.4.6</b> File reading practice</a></li>
<li class="chapter" data-level="10.4.7" data-path="exercises.html"><a href="exercises.html#bird-observations"><i class="fa fa-check"></i><b>10.4.7</b> Bird observations</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="exercises.html"><a href="exercises.html#functions-2"><i class="fa fa-check"></i><b>10.5</b> Functions</a><ul>
<li class="chapter" data-level="10.5.1" data-path="exercises.html"><a href="exercises.html#the-cut-function-i"><i class="fa fa-check"></i><b>10.5.1</b> The <code>cut()</code> function I</a></li>
<li class="chapter" data-level="10.5.2" data-path="exercises.html"><a href="exercises.html#the-cut-function-ii"><i class="fa fa-check"></i><b>10.5.2</b> The <code>cut()</code> function II</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="exercises.html"><a href="exercises.html#regular-expressions"><i class="fa fa-check"></i><b>10.6</b> Regular Expressions</a><ul>
<li class="chapter" data-level="10.6.1" data-path="exercises.html"><a href="exercises.html#restriction-enzymes-1"><i class="fa fa-check"></i><b>10.6.1</b> Restriction enzymes</a></li>
<li class="chapter" data-level="10.6.2" data-path="exercises.html"><a href="exercises.html#prosite-patterns-1"><i class="fa fa-check"></i><b>10.6.2</b> Prosite Patterns</a></li>
<li class="chapter" data-level="10.6.3" data-path="exercises.html"><a href="exercises.html#fasta-headers"><i class="fa fa-check"></i><b>10.6.3</b> Fasta Headers</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="exercises.html"><a href="exercises.html#scripting-2"><i class="fa fa-check"></i><b>10.7</b> Scripting</a><ul>
<li class="chapter" data-level="10.7.1" data-path="exercises.html"><a href="exercises.html#illegal-reproductions"><i class="fa fa-check"></i><b>10.7.1</b> Illegal reproductions</a></li>
<li class="chapter" data-level="10.7.2" data-path="exercises.html"><a href="exercises.html#count-odd-and-even-numbers"><i class="fa fa-check"></i><b>10.7.2</b> Count odd and even numbers</a></li>
<li class="chapter" data-level="10.7.3" data-path="exercises.html"><a href="exercises.html#add-column-compared-to-mean"><i class="fa fa-check"></i><b>10.7.3</b> Add column compared to mean</a></li>
<li class="chapter" data-level="10.7.4" data-path="exercises.html"><a href="exercises.html#interquantile-ranges"><i class="fa fa-check"></i><b>10.7.4</b> Interquantile ranges</a></li>
<li class="chapter" data-level="10.7.5" data-path="exercises.html"><a href="exercises.html#vector-distance"><i class="fa fa-check"></i><b>10.7.5</b> Vector distance</a></li>
<li class="chapter" data-level="10.7.6" data-path="exercises.html"><a href="exercises.html#quadratic-equation-solver"><i class="fa fa-check"></i><b>10.7.6</b> Quadratic equation solver</a></li>
<li class="chapter" data-level="10.7.7" data-path="exercises.html"><a href="exercises.html#gc-percentage-of-dna"><i class="fa fa-check"></i><b>10.7.7</b> G/C percentage of DNA</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="exercises.html"><a href="exercises.html#function-apply-and-its-relatives"><i class="fa fa-check"></i><b>10.8</b> Function <code>apply</code> and its relatives</a><ul>
<li class="chapter" data-level="10.8.1" data-path="exercises.html"><a href="exercises.html#whale-selenium"><i class="fa fa-check"></i><b>10.8.1</b> Whale selenium</a></li>
<li class="chapter" data-level="10.8.2" data-path="exercises.html"><a href="exercises.html#chickweight"><i class="fa fa-check"></i><b>10.8.2</b> ChickWeight</a></li>
<li class="chapter" data-level="10.8.3" data-path="exercises.html"><a href="exercises.html#food-constituents"><i class="fa fa-check"></i><b>10.8.3</b> Food constituents</a></li>
<li class="chapter" data-level="10.8.4" data-path="exercises.html"><a href="exercises.html#urine-properties"><i class="fa fa-check"></i><b>10.8.4</b> Urine properties</a></li>
<li class="chapter" data-level="10.8.5" data-path="exercises.html"><a href="exercises.html#bird-observations-revisited"><i class="fa fa-check"></i><b>10.8.5</b> Bird observations revisited</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="exercise-solutions.html"><a href="exercise-solutions.html"><i class="fa fa-check"></i><b>11</b> Exercise solutions</a><ul>
<li class="chapter" data-level="11.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#getting-started-2"><i class="fa fa-check"></i><b>11.1</b> Getting started</a><ul>
<li class="chapter" data-level="11.1.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#install-the-tools-1"><i class="fa fa-check"></i><b>11.1.1</b> Install the tools</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#toolbox-2"><i class="fa fa-check"></i><b>11.2</b> Toolbox</a><ul>
<li class="chapter" data-level="11.2.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#résumé-1"><i class="fa fa-check"></i><b>11.2.1</b> Résumé</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#basic-r-1"><i class="fa fa-check"></i><b>11.3</b> Basic R</a><ul>
<li class="chapter" data-level="11.3.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#math-in-the-console-1"><i class="fa fa-check"></i><b>11.3.1</b> Math in the console</a></li>
<li class="chapter" data-level="11.3.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#first-look-at-functions-1"><i class="fa fa-check"></i><b>11.3.2</b> First look at functions</a></li>
<li class="chapter" data-level="11.3.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#variables-2"><i class="fa fa-check"></i><b>11.3.3</b> Variables</a></li>
<li class="chapter" data-level="11.3.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#vectors-2"><i class="fa fa-check"></i><b>11.3.4</b> Vectors</a></li>
<li class="chapter" data-level="11.3.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#stair-walking-and-heart-rate-1"><i class="fa fa-check"></i><b>11.3.5</b> Stair walking and heart rate</a></li>
<li class="chapter" data-level="11.3.6" data-path="exercise-solutions.html"><a href="exercise-solutions.html#more-subjects-1"><i class="fa fa-check"></i><b>11.3.6</b> More subjects</a></li>
<li class="chapter" data-level="11.3.7" data-path="exercise-solutions.html"><a href="exercise-solutions.html#chickens-on-a-diet-1"><i class="fa fa-check"></i><b>11.3.7</b> Chickens on a diet</a></li>
<li class="chapter" data-level="11.3.8" data-path="exercise-solutions.html"><a href="exercise-solutions.html#chicken-bar-plot-1"><i class="fa fa-check"></i><b>11.3.8</b> Chicken bar plot</a></li>
<li class="chapter" data-level="11.3.9" data-path="exercise-solutions.html"><a href="exercise-solutions.html#discoveries-1"><i class="fa fa-check"></i><b>11.3.9</b> Discoveries</a></li>
<li class="chapter" data-level="11.3.10" data-path="exercise-solutions.html"><a href="exercise-solutions.html#lung-cancer-1"><i class="fa fa-check"></i><b>11.3.10</b> Lung cancer</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#complex-datatypes-1"><i class="fa fa-check"></i><b>11.4</b> Complex datatypes</a><ul>
<li class="chapter" data-level="11.4.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#creating-factors-1"><i class="fa fa-check"></i><b>11.4.1</b> Creating factors</a></li>
<li class="chapter" data-level="11.4.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#a-dictionary-with-a-named-vector-1"><i class="fa fa-check"></i><b>11.4.2</b> A dictionary with a named vector</a></li>
<li class="chapter" data-level="11.4.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#protein-concentrations-with-lowry-1"><i class="fa fa-check"></i><b>11.4.3</b> Protein concentrations with Lowry</a></li>
<li class="chapter" data-level="11.4.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#hplc-data-1"><i class="fa fa-check"></i><b>11.4.4</b> HPLC data</a></li>
<li class="chapter" data-level="11.4.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#airquality-1"><i class="fa fa-check"></i><b>11.4.5</b> Airquality</a></li>
<li class="chapter" data-level="11.4.6" data-path="exercise-solutions.html"><a href="exercise-solutions.html#file-reading-practice-1"><i class="fa fa-check"></i><b>11.4.6</b> File reading practice</a></li>
<li class="chapter" data-level="11.4.7" data-path="exercise-solutions.html"><a href="exercise-solutions.html#bird-observations-1"><i class="fa fa-check"></i><b>11.4.7</b> Bird observations</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#functions-3"><i class="fa fa-check"></i><b>11.5</b> Functions</a><ul>
<li class="chapter" data-level="11.5.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#the-cut-function-i-1"><i class="fa fa-check"></i><b>11.5.1</b> The <code>cut()</code> function I</a></li>
<li class="chapter" data-level="11.5.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#the-cut-function-ii-1"><i class="fa fa-check"></i><b>11.5.2</b> The <code>cut()</code> function II</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="exercise-solutions.html"><a href="exercise-solutions.html#regular-expressions-1"><i class="fa fa-check"></i><b>11.6</b> Regular Expressions</a><ul>
<li class="chapter" data-level="11.6.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#restriction-enzymes-2"><i class="fa fa-check"></i><b>11.6.1</b> Restriction enzymes</a></li>
<li class="chapter" data-level="11.6.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#prosite-patterns-2"><i class="fa fa-check"></i><b>11.6.2</b> Prosite Patterns</a></li>
<li class="chapter" data-level="11.6.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#fasta-headers-1"><i class="fa fa-check"></i><b>11.6.3</b> Fasta Headers</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="exercise-solutions.html"><a href="exercise-solutions.html#scripting-3"><i class="fa fa-check"></i><b>11.7</b> Scripting</a><ul>
<li class="chapter" data-level="11.7.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#illegal-reproductions-1"><i class="fa fa-check"></i><b>11.7.1</b> Illegal reproductions</a></li>
<li class="chapter" data-level="" data-path="exercise-solutions.html"><a href="exercise-solutions.html#find-match-locations-1"><i class="fa fa-check"></i>Find match locations</a></li>
<li class="chapter" data-level="11.7.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#count-odd-and-even-numbers-1"><i class="fa fa-check"></i><b>11.7.2</b> Count odd and even numbers</a></li>
<li class="chapter" data-level="11.7.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#add-column-compared-to-mean-1"><i class="fa fa-check"></i><b>11.7.3</b> Add column compared to mean</a></li>
<li class="chapter" data-level="11.7.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#interquantile-ranges-1"><i class="fa fa-check"></i><b>11.7.4</b> Interquantile ranges</a></li>
<li class="chapter" data-level="11.7.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#vector-distance-1"><i class="fa fa-check"></i><b>11.7.5</b> Vector distance</a></li>
<li class="chapter" data-level="11.7.6" data-path="exercise-solutions.html"><a href="exercise-solutions.html#quadratic-equation-solver-1"><i class="fa fa-check"></i><b>11.7.6</b> Quadratic equation solver</a></li>
<li class="chapter" data-level="11.7.7" data-path="exercise-solutions.html"><a href="exercise-solutions.html#gc-percentage-of-dna-1"><i class="fa fa-check"></i><b>11.7.7</b> G/C percentage of DNA</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="exercise-solutions.html"><a href="exercise-solutions.html#function-apply-and-its-relatives-1"><i class="fa fa-check"></i><b>11.8</b> Function <code>apply</code> and its relatives</a><ul>
<li class="chapter" data-level="11.8.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#whale-selenium-1"><i class="fa fa-check"></i><b>11.8.1</b> Whale selenium</a></li>
<li class="chapter" data-level="11.8.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#chickweight-1"><i class="fa fa-check"></i><b>11.8.2</b> ChickWeight</a></li>
<li class="chapter" data-level="11.8.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#food-constituents-1"><i class="fa fa-check"></i><b>11.8.3</b> Food constituents</a></li>
<li class="chapter" data-level="11.8.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#urine-properties-1"><i class="fa fa-check"></i><b>11.8.4</b> Urine properties</a></li>
<li class="chapter" data-level="11.8.5" data-path="exercise-solutions.html"><a href="exercise-solutions.html#bird-observations-revisited-1"><i class="fa fa-check"></i><b>11.8.5</b> Bird observations revisited</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis &amp; Visualization using R (1)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exploratory-data-analysis" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Exploratory Data Analysis</h1>
<div id="introduction-1" class="section level2">
<h2><span class="header-section-number">9.1</span> Introduction</h2>
<p>This chapter introduces you to the concept of Exploratory Data Analysis (EDA). In EDA a dataset is analysed with the goal of assessing its main characteristics, including its quality and usability for subsequent statistical modelling and analysis. Data summary techniques and visualizations ore used in an EDA. There is no fixed set of activities; each dataset poses its own questions and challenges.</p>
<p>Datasets are often expensive and time-consuming to collect, and therefore they are usually collected with a specific goal in mind. The EDA you perform should always be with this goal on the horizon: is the dataset of sufficient quality to answer the scientific questions for which they were collected?</p>
<div id="eda-outline" class="section level4 unnumbered">
<h4>EDA outline</h4>
<p>An EDA therefore may include:</p>
<ul>
<li><strong>Dataset description</strong>: how were they collected, what data types do the variables have, what are the units and physical quantities, how are missing data encoded, etc.</li>
<li><strong>Data summary</strong>: number of cases, number of variables, basic statistical description (i.e. mean, median, sd etc.), variable distributions (normal or not, outliers, missing data, skewed distributions)</li>
<li><strong>Visual summaries</strong> using boxplots, histograms, density plots.</li>
<li><strong>Recoding or transformations of variables</strong> may be required to get better results (e.g. from numeric to factor or vice versa, or log transformation of exponential data)</li>
<li><strong>Exploration of variable relations/covariance</strong>. It is always interesting to know about correlations between variables, but this is especially the case when you have a <em>dependent</em> variable for which you wish to build a statistical model in a later stage of your analysis.</li>
</ul>
<p>Note that your EDA results should <em>always</em> be accompanied by text that describes your results and discusses the implications of them. your figures should be well annotated with a caption and legend if relevant. An EDA is <em>not</em> a publication, but it will usually have a short introduction, a results section and a discussion. In contrast with a publication you usually <em>do</em> show the code. This ensures complete transparency and reproducibility.</p>
<p>In this chapter I will demonstrate a short EDA on the “Yeast dataset” from the UCI Machine Learning repository.</p>
<p>It is outside the scope of this short analysis to delve too deeply in the attribute background information, but you should realize that for a real analysis this is absolutely critical: without domain knowledge the you won’t have the insight to identify strange, worrying or surprising results.</p>
</div>
</div>
<div id="eda-of-the-yeast-dataset" class="section level2">
<h2><span class="header-section-number">9.2</span> EDA of the Yeast dataset</h2>
<div id="introduction-2" class="section level3">
<h3><span class="header-section-number">9.2.1</span> Introduction</h3>
<p>The data were collected from the <a href="https://archive.ics.uci.edu/ml/datasets/Yeast">UCI Machine Learning website</a>. To ensure their continued availability the files were copied to a personal <a href="https://github.com/MichielNoback/datasets/tree/master/UCI_yeast_protein_loc">repository</a>.</p>
<p>The data were accompanied by a <em>very</em> short abstract: <strong><em>“Predicting the Cellular Localization Sites of Proteins”</em></strong>. The data description states there are 1484 instances, each with 8 attributes (variables) and a class label. The class label is the variable we wish to build a model for so this is the dependent variable.</p>
<p>The <em>Attribute Information</em> section describes these variables:</p>
<ol style="list-style-type: decimal">
<li>Sequence Name: Accession number for the SWISS-PROT database</li>
<li><code>mcg</code>: McGeoch’s method for signal sequence recognition.</li>
<li><code>gvh</code>: von Heijne’s method for signal sequence recognition.</li>
<li><code>alm</code>: Score of the ALOM membrane spanning region prediction program.</li>
<li><code>mit</code>: Score of discriminant analysis of the amino acid content of the N-terminal region (20 residues long) of mitochondrial and non-mitochondrial proteins.</li>
<li><code>erl</code>: Presence of “HDEL” substring (thought to act as a signal for retention in the endoplasmic reticulum lumen). Binary attribute.</li>
<li><code>pox</code>: Peroxisomal targeting signal in the C-terminus.</li>
<li><code>vac</code>: Score of discriminant analysis of the amino acid content of vacuolar and extracellular proteins.</li>
<li><code>nuc</code>: Score of discriminant analysis of nuclear localization signals of nuclear and non-nuclear proteins.</li>
</ol>
<p>The sequence name is not interesting in this EDA: we are interested in patterns, not individuals. However, if there is an anomalous protein in the EDA it should be possible to retrace its origin so the identifier stays in the dataset.</p>
<p>There seem to be two attributes (<code>mcg</code> and <code>gvh</code>) measuring the same property - the presence of a signal sequence which is the N-terminal part of a protein signalling the cellular machinery the protein should be exported.</p>
<p>Very simply put, if this scoring system was perfect each of the attributes (except <code>mcg</code> and <code>gvh</code>) would unequivocally “assign” the protein in question to a cellular location: Cellular external matrix, membrane inserted, mitochondrial, endoplasmic reticulum lumen, peroxisomal, vacuolar or nuclear.</p>
<p>The tenth variable in the data is the dependent or explanatory variable. The <code>yeast.names</code> file describes it like this:</p>
<pre><code>Class Distribution. The class is the localization site. Please see Nakai &amp;
               Kanehisa referenced above for more details.
  CYT (cytosolic or cytoskeletal)                    463
  NUC (nuclear)                                      429
  MIT (mitochondrial)                                244
  ME3 (membrane protein, no N-terminal signal)       163
  ME2 (membrane protein, uncleaved signal)            51
  ME1 (membrane protein, cleaved signal)              44
  EXC (extracellular)                                 37
  VAC (vacuolar)                                      30
  POX (peroxisomal)                                   20
  ERL (endoplasmic reticulum lumen)                    5</code></pre>
<p>This tells me the different localizations are by no means equally distributed; there is an over representation of “cytosolic or cytoskeletal” and “nuclear” and a huge under representation of especially endoplasmic reticulum lumen proteins.</p>
</div>
<div id="data-loading-and-prepping" class="section level3">
<h3><span class="header-section-number">9.2.2</span> Data loading and prepping</h3>
<p>Since I am going to rerun the code in this notebook often I am going to create a local copy, and load that one.</p>
<div class="sourceCode" id="cb489"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb489-1"><a href="exploratory-data-analysis.html#cb489-1"></a>file_name &lt;-<span class="st"> &quot;yeast.data&quot;</span></span>
<span id="cb489-2"><a href="exploratory-data-analysis.html#cb489-2"></a>yeast_data_url &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;https://raw.githubusercontent.com/MichielNoback/datasets/master/UCI_yeast_protein_loc/&quot;</span>, file_name)</span>
<span id="cb489-3"><a href="exploratory-data-analysis.html#cb489-3"></a>yeast_local_location &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;../&quot;</span>, file_name)</span>
<span id="cb489-4"><a href="exploratory-data-analysis.html#cb489-4"></a></span>
<span id="cb489-5"><a href="exploratory-data-analysis.html#cb489-5"></a><span class="co">#only download if not present</span></span>
<span id="cb489-6"><a href="exploratory-data-analysis.html#cb489-6"></a><span class="cf">if</span> (<span class="op">!</span><span class="st"> </span><span class="kw">file.exists</span>(yeast_local_location)) {</span>
<span id="cb489-7"><a href="exploratory-data-analysis.html#cb489-7"></a>    <span class="kw">download.file</span>(<span class="dt">url =</span> yeast_data_url, </span>
<span id="cb489-8"><a href="exploratory-data-analysis.html#cb489-8"></a>                  <span class="dt">destfile =</span> yeast_local_location)</span>
<span id="cb489-9"><a href="exploratory-data-analysis.html#cb489-9"></a>}</span>
<span id="cb489-10"><a href="exploratory-data-analysis.html#cb489-10"></a></span>
<span id="cb489-11"><a href="exploratory-data-analysis.html#cb489-11"></a>yeast_data &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">file =</span> yeast_local_location,</span>
<span id="cb489-12"><a href="exploratory-data-analysis.html#cb489-12"></a>                         <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>,</span>
<span id="cb489-13"><a href="exploratory-data-analysis.html#cb489-13"></a>                         <span class="dt">as.is =</span> <span class="dv">1</span>)</span>
<span id="cb489-14"><a href="exploratory-data-analysis.html#cb489-14"></a><span class="kw">str</span>(yeast_data)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    1484 obs. of  10 variables:
##  $ V1 : chr  &quot;ADT1_YEAST&quot; &quot;ADT2_YEAST&quot; &quot;ADT3_YEAST&quot; &quot;AAR2_YEAST&quot; ...
##  $ V2 : num  0.58 0.43 0.64 0.58 0.42 0.51 0.5 0.48 0.55 0.4 ...
##  $ V3 : num  0.61 0.67 0.62 0.44 0.44 0.4 0.54 0.45 0.5 0.39 ...
##  $ V4 : num  0.47 0.48 0.49 0.57 0.48 0.56 0.48 0.59 0.66 0.6 ...
##  $ V5 : num  0.13 0.27 0.15 0.13 0.54 0.17 0.65 0.2 0.36 0.15 ...
##  $ V6 : num  0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 ...
##  $ V7 : num  0 0 0 0 0 0.5 0 0 0 0 ...
##  $ V8 : num  0.48 0.53 0.53 0.54 0.48 0.49 0.53 0.58 0.49 0.58 ...
##  $ V9 : num  0.22 0.22 0.22 0.22 0.22 0.22 0.22 0.34 0.22 0.3 ...
##  $ V10: Factor w/ 10 levels &quot;CYT&quot;,&quot;ERL&quot;,&quot;EXC&quot;,..: 7 7 7 8 7 1 7 8 7 1 ...</code></pre>
<p>The data seems to have been loaded correctly and, as expected since that was stated in the original data description, there are no missing data.</p>
<p>The column names were not defined in the data file so this will be fixed first. I will create a data frame that also holds the column descriptions. I have put the data in a small text file for easy loading and editing; the attribute descriptions copied/pasted into a text file and with find/replace converted in easy to load form, and a new <code>label</code> column added for use in plotting. There were a few occurrences of the <code>'</code> character which always corrupt data import into R. They were removed.</p>
<div class="sourceCode" id="cb491"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb491-1"><a href="exploratory-data-analysis.html#cb491-1"></a>attribute_info &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/yeast_attribute_info.txt&quot;</span>, </span>
<span id="cb491-2"><a href="exploratory-data-analysis.html#cb491-2"></a>                             <span class="dt">sep =</span> <span class="st">&quot;:&quot;</span>,</span>
<span id="cb491-3"><a href="exploratory-data-analysis.html#cb491-3"></a>                             <span class="dt">header =</span> <span class="ot">TRUE</span>,</span>
<span id="cb491-4"><a href="exploratory-data-analysis.html#cb491-4"></a>                             <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb491-5"><a href="exploratory-data-analysis.html#cb491-5"></a><span class="co">#attach column names</span></span>
<span id="cb491-6"><a href="exploratory-data-analysis.html#cb491-6"></a>(<span class="kw">colnames</span>(yeast_data) &lt;-<span class="st"> </span>attribute_info<span class="op">$</span>attribute)</span></code></pre></div>
<pre><code>##  [1] &quot;accno&quot; &quot;mcg&quot;   &quot;gvh&quot;   &quot;alm&quot;   &quot;mit&quot;   &quot;erl&quot;   &quot;pox&quot;   &quot;vac&quot;   &quot;nuc&quot;  
## [10] &quot;loc&quot;</code></pre>
<p>To make this info easily accessible, a function is created that can be used to fetch either the description or the label.</p>
<div class="sourceCode" id="cb493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb493-1"><a href="exploratory-data-analysis.html#cb493-1"></a>get_attribute_info &lt;-<span class="st"> </span><span class="cf">function</span>(attribute, <span class="dt">resource =</span> <span class="st">&quot;label&quot;</span>) {</span>
<span id="cb493-2"><a href="exploratory-data-analysis.html#cb493-2"></a>    <span class="cf">if</span> (<span class="op">!</span><span class="st"> </span>resource <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;label&quot;</span>, <span class="st">&quot;description&quot;</span>)) {</span>
<span id="cb493-3"><a href="exploratory-data-analysis.html#cb493-3"></a>        <span class="kw">stop</span>(<span class="kw">paste0</span>(<span class="st">&quot;type &quot;</span>, resource, <span class="st">&quot;is not an attribute resource&quot;</span>))</span>
<span id="cb493-4"><a href="exploratory-data-analysis.html#cb493-4"></a>    }</span>
<span id="cb493-5"><a href="exploratory-data-analysis.html#cb493-5"></a>    <span class="kw">return</span>(attribute_info[attribute_info<span class="op">$</span>attribute <span class="op">==</span><span class="st"> </span>attribute, resource])</span>
<span id="cb493-6"><a href="exploratory-data-analysis.html#cb493-6"></a>}</span>
<span id="cb493-7"><a href="exploratory-data-analysis.html#cb493-7"></a></span>
<span id="cb493-8"><a href="exploratory-data-analysis.html#cb493-8"></a><span class="co">#test it</span></span>
<span id="cb493-9"><a href="exploratory-data-analysis.html#cb493-9"></a><span class="kw">get_attribute_info</span>(<span class="st">&quot;gvh&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;VonHeijne signal&quot;</code></pre>
<div class="sourceCode" id="cb495"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb495-1"><a href="exploratory-data-analysis.html#cb495-1"></a><span class="kw">get_attribute_info</span>(<span class="st">&quot;accno&quot;</span>, <span class="st">&quot;description&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Accession number for the SWISS-PROT database&quot;</code></pre>
</div>
<div id="data-verification" class="section level3">
<h3><span class="header-section-number">9.2.3</span> Data verification</h3>
<p>The original data description stated that 1484 instances with 9 attributes + a class label are present.</p>
<div class="sourceCode" id="cb497"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb497-1"><a href="exploratory-data-analysis.html#cb497-1"></a><span class="kw">dim</span>(yeast_data)</span></code></pre></div>
<pre><code>## [1] 1484   10</code></pre>
<p>This is correct.
No missing data are supposed to be there:</p>
<div class="sourceCode" id="cb499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb499-1"><a href="exploratory-data-analysis.html#cb499-1"></a><span class="kw">sum</span>(<span class="kw">complete.cases</span>(yeast_data))</span></code></pre></div>
<pre><code>## [1] 1484</code></pre>
<p>Also correct.</p>
<p>The classes of the columns are also OK so the data is verified and found correct.</p>
</div>
<div id="attribute-summaries" class="section level3">
<h3><span class="header-section-number">9.2.4</span> Attribute summaries</h3>
<p>A first scan of the attributes.</p>
<div class="sourceCode" id="cb501"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb501-1"><a href="exploratory-data-analysis.html#cb501-1"></a><span class="kw">summary</span>(yeast_data)</span></code></pre></div>
<pre><code>##     accno                mcg            gvh            alm      
##  Length:1484        Min.   :0.11   Min.   :0.13   Min.   :0.21  
##  Class :character   1st Qu.:0.41   1st Qu.:0.42   1st Qu.:0.46  
##  Mode  :character   Median :0.49   Median :0.49   Median :0.51  
##                     Mean   :0.50   Mean   :0.50   Mean   :0.50  
##                     3rd Qu.:0.58   3rd Qu.:0.57   3rd Qu.:0.55  
##                     Max.   :1.00   Max.   :1.00   Max.   :1.00  
##                                                                 
##       mit             erl             pox              vac      
##  Min.   :0.000   Min.   :0.500   Min.   :0.0000   Min.   :0.00  
##  1st Qu.:0.170   1st Qu.:0.500   1st Qu.:0.0000   1st Qu.:0.48  
##  Median :0.220   Median :0.500   Median :0.0000   Median :0.51  
##  Mean   :0.261   Mean   :0.505   Mean   :0.0075   Mean   :0.50  
##  3rd Qu.:0.320   3rd Qu.:0.500   3rd Qu.:0.0000   3rd Qu.:0.53  
##  Max.   :1.000   Max.   :1.000   Max.   :0.8300   Max.   :0.73  
##                                                                 
##       nuc             loc     
##  Min.   :0.000   CYT    :463  
##  1st Qu.:0.220   NUC    :429  
##  Median :0.220   MIT    :244  
##  Mean   :0.276   ME3    :163  
##  3rd Qu.:0.300   ME2    : 51  
##  Max.   :1.000   ME1    : 44  
##                  (Other): 90</code></pre>
<p>All these attributes seem to be in the range zero to one, except of course for the localization attribute and the accession number. This is not surprising since these attributes are all probabilities. The class distribution corresponds with the published data.</p>
<p>The
The <code>pox</code> (Peroxisomal) and <code>erl</code> ( ER-retention) attributes, seem to have a really strange distribution and should be investigated further.</p>
<p>Here are all the numeric attributes in a single panel. I chose histogram over boxplot or density plot because it is more fine-grained than boxplot and very easy to interpret.</p>
<div class="sourceCode" id="cb503"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb503-1"><a href="exploratory-data-analysis.html#cb503-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb503-2"><a href="exploratory-data-analysis.html#cb503-2"></a></span>
<span id="cb503-3"><a href="exploratory-data-analysis.html#cb503-3"></a><span class="cf">for</span> ( attrib <span class="cf">in</span> <span class="kw">colnames</span>(yeast_data[<span class="dv">2</span><span class="op">:</span><span class="dv">9</span>])) {</span>
<span id="cb503-4"><a href="exploratory-data-analysis.html#cb503-4"></a>    <span class="kw">hist</span>(yeast_data[, attrib],</span>
<span id="cb503-5"><a href="exploratory-data-analysis.html#cb503-5"></a>         <span class="dt">main =</span> <span class="ot">NULL</span>,</span>
<span id="cb503-6"><a href="exploratory-data-analysis.html#cb503-6"></a>         <span class="dt">xlab =</span> <span class="kw">paste0</span>(<span class="kw">get_attribute_info</span>(attrib), <span class="st">&quot; (prob)&quot;</span>))</span>
<span id="cb503-7"><a href="exploratory-data-analysis.html#cb503-7"></a>}</span></code></pre></div>
<div class="figure"><span id="fig:yeast-attributes-hist"></span>
<img src="davur_ebook_files/figure-html/yeast-attributes-hist-1.png" alt="Distributions of the numeric variables" width="100%" />
<p class="caption">
Figure 9.1: Distributions of the numeric variables
</p>
</div>
<div class="sourceCode" id="cb504"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb504-1"><a href="exploratory-data-analysis.html#cb504-1"></a><span class="co">#reset par()</span></span>
<span id="cb504-2"><a href="exploratory-data-analysis.html#cb504-2"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span></code></pre></div>
<p>Here, it can be seen that the attributes “ER-retention” and “Peroxisomal” are pretty much uninformative, a problem probably largely caused by the low abundance of proteins in these classes. Surprisingly enough, the “Vacuolar” property which is also low-abundant, does not have this extreme distribution.</p>
<p>What is also striking is that, although these attributes are probabilities of targeting signals, none (with the exception of the two low-abundance ones) show a more or less bi-modal distribution as you would naively expect.</p>
</div>
<div id="attribute-relationships" class="section level3">
<h3><span class="header-section-number">9.2.5</span> Attribute relationships</h3>
<p>Here is a quick scan of variable relationship, excluding the dependent variable. The <code>pairs()</code> function is used for that. I excluded <code>erl</code> and <code>pox</code> because they do not add information to the picture.</p>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="exploratory-data-analysis.html#cb505-1"></a><span class="kw">pairs</span>(yeast_data[<span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">8</span>, <span class="dv">9</span>)],</span>
<span id="cb505-2"><a href="exploratory-data-analysis.html#cb505-2"></a>      <span class="dt">panel =</span> <span class="st">&quot;panel.smooth&quot;</span>,</span>
<span id="cb505-3"><a href="exploratory-data-analysis.html#cb505-3"></a>      <span class="dt">pch =</span> <span class="dv">20</span>, </span>
<span id="cb505-4"><a href="exploratory-data-analysis.html#cb505-4"></a>      <span class="dt">cex =</span> <span class="fl">0.35</span>,</span>
<span id="cb505-5"><a href="exploratory-data-analysis.html#cb505-5"></a>      <span class="dt">col =</span> <span class="kw">rgb</span>(<span class="fl">0.1</span>, <span class="fl">0.4</span>, <span class="dv">1</span>, <span class="dt">alpha =</span> <span class="fl">0.4</span>))</span></code></pre></div>
<div class="figure"><span id="fig:yeast-pairs-plot"></span>
<img src="davur_ebook_files/figure-html/yeast-pairs-plot-1.png" alt="Relations between the numeric variables" width="100%" />
<p class="caption">
Figure 9.2: Relations between the numeric variables
</p>
</div>
<p>By adding the smoother it is made clear that the only pair which shows a clear correlation is the pair <code>mcg</code> / <code>gvh</code> which actually predict the same property. Therefore it would have been very surprising indeed if they would <em>not</em> have a correlation. The strength of this correlation -the <strong><em>R-squared</em></strong>- is this</p>
<div class="sourceCode" id="cb506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb506-1"><a href="exploratory-data-analysis.html#cb506-1"></a>lin_mod &lt;-<span class="st"> </span><span class="kw">lm</span>(yeast_data<span class="op">$</span>mcg <span class="op">~</span><span class="st"> </span>yeast_data<span class="op">$</span>gvh)</span>
<span id="cb506-2"><a href="exploratory-data-analysis.html#cb506-2"></a><span class="kw">summary</span>(lin_mod)<span class="op">$</span>r.squared</span></code></pre></div>
<pre><code>## [1] 0.3383</code></pre>
<p>This is not very strong.</p>
</div>
<div id="correlations-to-the-dependent-variable" class="section level3">
<h3><span class="header-section-number">9.2.6</span> Correlations to the dependent variable</h3>
<p>As a first investigation, numeric variables are split on the location variable.</p>
<div class="sourceCode" id="cb508"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb508-1"><a href="exploratory-data-analysis.html#cb508-1"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb508-2"><a href="exploratory-data-analysis.html#cb508-2"></a>colors &lt;-<span class="st"> </span>RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="dt">name =</span> <span class="st">&quot;Accent&quot;</span>)</span>
<span id="cb508-3"><a href="exploratory-data-analysis.html#cb508-3"></a><span class="co">#col = c(&quot;darkblue&quot;, &quot;darkgreen&quot;, &quot;magenta&quot;, )</span></span>
<span id="cb508-4"><a href="exploratory-data-analysis.html#cb508-4"></a>col_i &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb508-5"><a href="exploratory-data-analysis.html#cb508-5"></a><span class="cf">for</span> (name <span class="cf">in</span> <span class="kw">names</span>(yeast_data[<span class="dv">2</span><span class="op">:</span><span class="dv">9</span>])) {</span>
<span id="cb508-6"><a href="exploratory-data-analysis.html#cb508-6"></a>    col_i &lt;-<span class="st"> </span>col_i <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb508-7"><a href="exploratory-data-analysis.html#cb508-7"></a>    <span class="kw">boxplot</span>(yeast_data[, name] <span class="op">~</span><span class="st"> </span>yeast_data<span class="op">$</span>loc,</span>
<span id="cb508-8"><a href="exploratory-data-analysis.html#cb508-8"></a>            <span class="dt">xlab =</span> <span class="ot">NULL</span>,</span>
<span id="cb508-9"><a href="exploratory-data-analysis.html#cb508-9"></a>            <span class="dt">ylab =</span> <span class="kw">get_attribute_info</span>(name),</span>
<span id="cb508-10"><a href="exploratory-data-analysis.html#cb508-10"></a>            <span class="dt">col =</span> colors[col_i])</span>
<span id="cb508-11"><a href="exploratory-data-analysis.html#cb508-11"></a>}</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:corr-to-dependent-1"></span>
<img src="davur_ebook_files/figure-html/corr-to-dependent-1.png" alt="Correlations with the dependent variable" width="70%" />
<p class="caption">
Figure 9.3: Correlations with the dependent variable
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:corr-to-dependent-2"></span>
<img src="davur_ebook_files/figure-html/corr-to-dependent-2.png" alt="Correlations with the dependent variable" width="70%" />
<p class="caption">
Figure 9.4: Correlations with the dependent variable
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:corr-to-dependent-3"></span>
<img src="davur_ebook_files/figure-html/corr-to-dependent-3.png" alt="Correlations with the dependent variable" width="70%" />
<p class="caption">
Figure 9.5: Correlations with the dependent variable
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:corr-to-dependent-4"></span>
<img src="davur_ebook_files/figure-html/corr-to-dependent-4.png" alt="Correlations with the dependent variable" width="70%" />
<p class="caption">
Figure 9.6: Correlations with the dependent variable
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:corr-to-dependent-5"></span>
<img src="davur_ebook_files/figure-html/corr-to-dependent-5.png" alt="Correlations with the dependent variable" width="70%" />
<p class="caption">
Figure 9.7: Correlations with the dependent variable
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:corr-to-dependent-6"></span>
<img src="davur_ebook_files/figure-html/corr-to-dependent-6.png" alt="Correlations with the dependent variable" width="70%" />
<p class="caption">
Figure 9.8: Correlations with the dependent variable
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:corr-to-dependent-7"></span>
<img src="davur_ebook_files/figure-html/corr-to-dependent-7.png" alt="Correlations with the dependent variable" width="70%" />
<p class="caption">
Figure 9.9: Correlations with the dependent variable
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:corr-to-dependent-8"></span>
<img src="davur_ebook_files/figure-html/corr-to-dependent-8.png" alt="Correlations with the dependent variable" width="70%" />
<p class="caption">
Figure 9.10: Correlations with the dependent variable
</p>
</div>
<p>Quite a few of the variables correlate pretty well with the dependent variable. These do so very well: <code>alm</code>, <code>mit</code>, <code>erl</code>, <code>pox</code>. Others are not: <code>nuc</code>, <code>vac</code>, and some are ambiguous - they discriminate, but not exclusively: <code>mcg</code>, <code>gvh</code>.</p>
<p>Here are some summary statistics for the numeric variables split on the dependent variable:</p>
<div class="sourceCode" id="cb509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb509-1"><a href="exploratory-data-analysis.html#cb509-1"></a><span class="kw">options</span>(<span class="dt">digits=</span><span class="dv">2</span>)</span>
<span id="cb509-2"><a href="exploratory-data-analysis.html#cb509-2"></a></span>
<span id="cb509-3"><a href="exploratory-data-analysis.html#cb509-3"></a>tmp &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">aggregate</span>( . <span class="op">~</span><span class="st"> </span>loc, <span class="dt">data =</span> yeast_data[, <span class="dv">-1</span>], <span class="dt">FUN =</span> mean))</span>
<span id="cb509-4"><a href="exploratory-data-analysis.html#cb509-4"></a>knitr<span class="op">::</span><span class="kw">kable</span>(tmp)</span></code></pre></div>
<pre><code>## Warning in kable_pipe(x = structure(c(&quot;loc&quot;, &quot;mcg&quot;, &quot;gvh&quot;, &quot;alm&quot;, &quot;mit&quot;, : The
## table should have a header (column names)</code></pre>
<table>
<tbody>
<tr class="odd">
<td align="left">loc</td>
<td align="left">CYT</td>
<td align="left">ERL</td>
<td align="left">EXC</td>
<td align="left">ME1</td>
<td align="left">ME2</td>
<td align="left">ME3</td>
<td align="left">MIT</td>
<td align="left">NUC</td>
<td align="left">POX</td>
<td align="left">VAC</td>
</tr>
<tr class="even">
<td align="left">mcg</td>
<td align="left">0.48</td>
<td align="left">0.79</td>
<td align="left">0.74</td>
<td align="left">0.79</td>
<td align="left">0.72</td>
<td align="left">0.43</td>
<td align="left">0.52</td>
<td align="left">0.45</td>
<td align="left">0.52</td>
<td align="left">0.55</td>
</tr>
<tr class="odd">
<td align="left">gvh</td>
<td align="left">0.47</td>
<td align="left">0.77</td>
<td align="left">0.72</td>
<td align="left">0.76</td>
<td align="left">0.60</td>
<td align="left">0.49</td>
<td align="left">0.53</td>
<td align="left">0.46</td>
<td align="left">0.51</td>
<td align="left">0.53</td>
</tr>
<tr class="even">
<td align="left">alm</td>
<td align="left">0.54</td>
<td align="left">0.48</td>
<td align="left">0.49</td>
<td align="left">0.38</td>
<td align="left">0.41</td>
<td align="left">0.36</td>
<td align="left">0.52</td>
<td align="left">0.53</td>
<td align="left">0.51</td>
<td align="left">0.47</td>
</tr>
<tr class="odd">
<td align="left">mit</td>
<td align="left">0.23</td>
<td align="left">0.34</td>
<td align="left">0.29</td>
<td align="left">0.31</td>
<td align="left">0.28</td>
<td align="left">0.21</td>
<td align="left">0.40</td>
<td align="left">0.23</td>
<td align="left">0.25</td>
<td align="left">0.20</td>
</tr>
<tr class="even">
<td align="left">erl</td>
<td align="left">0.50</td>
<td align="left">1.00</td>
<td align="left">0.50</td>
<td align="left">0.50</td>
<td align="left">0.51</td>
<td align="left">0.50</td>
<td align="left">0.50</td>
<td align="left">0.50</td>
<td align="left">0.50</td>
<td align="left">0.50</td>
</tr>
<tr class="odd">
<td align="left">pox</td>
<td align="left">0.0011</td>
<td align="left">0.0000</td>
<td align="left">0.0000</td>
<td align="left">0.0000</td>
<td align="left">0.0000</td>
<td align="left">0.0000</td>
<td align="left">0.0089</td>
<td align="left">0.0000</td>
<td align="left">0.4235</td>
<td align="left">0.0000</td>
</tr>
<tr class="even">
<td align="left">vac</td>
<td align="left">0.50</td>
<td align="left">0.55</td>
<td align="left">0.46</td>
<td align="left">0.51</td>
<td align="left">0.51</td>
<td align="left">0.51</td>
<td align="left">0.50</td>
<td align="left">0.49</td>
<td align="left">0.50</td>
<td align="left">0.53</td>
</tr>
<tr class="odd">
<td align="left">nuc</td>
<td align="left">0.26</td>
<td align="left">0.25</td>
<td align="left">0.23</td>
<td align="left">0.27</td>
<td align="left">0.25</td>
<td align="left">0.27</td>
<td align="left">0.24</td>
<td align="left">0.33</td>
<td align="left">0.23</td>
<td align="left">0.25</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb511-1"><a href="exploratory-data-analysis.html#cb511-1"></a>tmp &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">aggregate</span>( . <span class="op">~</span><span class="st"> </span>loc, <span class="dt">data =</span> yeast_data[, <span class="dv">-1</span>], <span class="dt">FUN =</span> median))</span>
<span id="cb511-2"><a href="exploratory-data-analysis.html#cb511-2"></a>knitr<span class="op">::</span><span class="kw">kable</span>(tmp)</span></code></pre></div>
<pre><code>## Warning in kable_pipe(x = structure(c(&quot;loc&quot;, &quot;mcg&quot;, &quot;gvh&quot;, &quot;alm&quot;, &quot;mit&quot;, : The
## table should have a header (column names)</code></pre>
<table>
<tbody>
<tr class="odd">
<td align="left">loc</td>
<td align="left">CYT</td>
<td align="left">ERL</td>
<td align="left">EXC</td>
<td align="left">ME1</td>
<td align="left">ME2</td>
<td align="left">ME3</td>
<td align="left">MIT</td>
<td align="left">NUC</td>
<td align="left">POX</td>
<td align="left">VAC</td>
</tr>
<tr class="even">
<td align="left">mcg</td>
<td align="left">0.48</td>
<td align="left">0.78</td>
<td align="left">0.77</td>
<td align="left">0.78</td>
<td align="left">0.76</td>
<td align="left">0.43</td>
<td align="left">0.51</td>
<td align="left">0.45</td>
<td align="left">0.48</td>
<td align="left">0.56</td>
</tr>
<tr class="odd">
<td align="left">gvh</td>
<td align="left">0.46</td>
<td align="left">0.80</td>
<td align="left">0.73</td>
<td align="left">0.73</td>
<td align="left">0.59</td>
<td align="left">0.49</td>
<td align="left">0.53</td>
<td align="left">0.46</td>
<td align="left">0.51</td>
<td align="left">0.51</td>
</tr>
<tr class="even">
<td align="left">alm</td>
<td align="left">0.53</td>
<td align="left">0.50</td>
<td align="left">0.49</td>
<td align="left">0.36</td>
<td align="left">0.41</td>
<td align="left">0.36</td>
<td align="left">0.52</td>
<td align="left">0.53</td>
<td align="left">0.51</td>
<td align="left">0.49</td>
</tr>
<tr class="odd">
<td align="left">mit</td>
<td align="left">0.19</td>
<td align="left">0.37</td>
<td align="left">0.28</td>
<td align="left">0.29</td>
<td align="left">0.26</td>
<td align="left">0.20</td>
<td align="left">0.41</td>
<td align="left">0.20</td>
<td align="left">0.21</td>
<td align="left">0.17</td>
</tr>
<tr class="even">
<td align="left">erl</td>
<td align="left">0.5</td>
<td align="left">1.0</td>
<td align="left">0.5</td>
<td align="left">0.5</td>
<td align="left">0.5</td>
<td align="left">0.5</td>
<td align="left">0.5</td>
<td align="left">0.5</td>
<td align="left">0.5</td>
<td align="left">0.5</td>
</tr>
<tr class="odd">
<td align="left">pox</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
<td align="left">0.5</td>
<td align="left">0.0</td>
</tr>
<tr class="even">
<td align="left">vac</td>
<td align="left">0.51</td>
<td align="left">0.54</td>
<td align="left">0.48</td>
<td align="left">0.53</td>
<td align="left">0.52</td>
<td align="left">0.51</td>
<td align="left">0.50</td>
<td align="left">0.50</td>
<td align="left">0.51</td>
<td align="left">0.53</td>
</tr>
<tr class="odd">
<td align="left">nuc</td>
<td align="left">0.22</td>
<td align="left">0.22</td>
<td align="left">0.22</td>
<td align="left">0.22</td>
<td align="left">0.22</td>
<td align="left">0.22</td>
<td align="left">0.22</td>
<td align="left">0.28</td>
<td align="left">0.22</td>
<td align="left">0.22</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dataframe-manipulations-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="exercises.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["davur_ebook.pdf", "davur_ebook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
